{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Importing Libraries and Modules\n",
    "import os\n",
    "import re\n",
    "import xml.etree.ElementTree as ET\n",
    "from bs4 import BeautifulSoup\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting de-core-news-lg==3.7.0\n",
      "  Downloading https://github.com/explosion/spacy-models/releases/download/de_core_news_lg-3.7.0/de_core_news_lg-3.7.0-py3-none-any.whl (567.8 MB)\n",
      "     ---------------------------------------- 0.0/567.8 MB ? eta -:--:--\n",
      "     ---------------------------------------- 0.3/567.8 MB 7.9 MB/s eta 0:01:12\n",
      "     ---------------------------------------- 0.7/567.8 MB 9.3 MB/s eta 0:01:01\n",
      "     --------------------------------------- 2.2/567.8 MB 17.2 MB/s eta 0:00:33\n",
      "     --------------------------------------- 3.9/567.8 MB 22.5 MB/s eta 0:00:26\n",
      "     --------------------------------------- 4.9/567.8 MB 22.6 MB/s eta 0:00:25\n",
      "     --------------------------------------- 5.8/567.8 MB 23.2 MB/s eta 0:00:25\n",
      "     --------------------------------------- 5.8/567.8 MB 23.2 MB/s eta 0:00:25\n",
      "      -------------------------------------- 8.6/567.8 MB 23.8 MB/s eta 0:00:24\n",
      "      ------------------------------------- 10.3/567.8 MB 25.2 MB/s eta 0:00:23\n",
      "      ------------------------------------- 11.4/567.8 MB 28.4 MB/s eta 0:00:20\n",
      "      ------------------------------------- 13.2/567.8 MB 29.7 MB/s eta 0:00:19\n",
      "      ------------------------------------- 14.4/567.8 MB 29.7 MB/s eta 0:00:19\n",
      "     - ------------------------------------ 15.9/567.8 MB 29.8 MB/s eta 0:00:19\n",
      "     - ------------------------------------ 17.4/567.8 MB 34.4 MB/s eta 0:00:16\n",
      "     - ------------------------------------ 18.6/567.8 MB 31.1 MB/s eta 0:00:18\n",
      "     - ------------------------------------ 20.0/567.8 MB 29.7 MB/s eta 0:00:19\n",
      "     - ------------------------------------ 21.4/567.8 MB 31.2 MB/s eta 0:00:18\n",
      "     - ------------------------------------ 22.9/567.8 MB 29.7 MB/s eta 0:00:19\n",
      "     - ------------------------------------ 24.1/567.8 MB 29.7 MB/s eta 0:00:19\n",
      "     - ------------------------------------ 25.3/567.8 MB 29.7 MB/s eta 0:00:19\n",
      "     - ------------------------------------ 26.3/567.8 MB 28.5 MB/s eta 0:00:20\n",
      "     - ------------------------------------ 27.7/567.8 MB 28.4 MB/s eta 0:00:20\n",
      "     - ------------------------------------ 28.8/567.8 MB 26.2 MB/s eta 0:00:21\n",
      "     - ------------------------------------ 29.8/567.8 MB 26.2 MB/s eta 0:00:21\n",
      "     -- ----------------------------------- 30.9/567.8 MB 27.3 MB/s eta 0:00:20\n",
      "     -- ----------------------------------- 31.5/567.8 MB 24.2 MB/s eta 0:00:23\n",
      "     -- ----------------------------------- 33.5/567.8 MB 26.2 MB/s eta 0:00:21\n",
      "     -- ----------------------------------- 34.7/567.8 MB 25.2 MB/s eta 0:00:22\n",
      "     -- ----------------------------------- 35.6/567.8 MB 25.2 MB/s eta 0:00:22\n",
      "     -- ----------------------------------- 36.6/567.8 MB 25.1 MB/s eta 0:00:22\n",
      "     -- ----------------------------------- 38.0/567.8 MB 25.1 MB/s eta 0:00:22\n",
      "     -- ----------------------------------- 38.9/567.8 MB 25.2 MB/s eta 0:00:21\n",
      "     -- ----------------------------------- 40.2/567.8 MB 25.2 MB/s eta 0:00:21\n",
      "     -- ----------------------------------- 41.7/567.8 MB 28.5 MB/s eta 0:00:19\n",
      "     -- ----------------------------------- 43.0/567.8 MB 26.2 MB/s eta 0:00:21\n",
      "     -- ----------------------------------- 44.2/567.8 MB 26.2 MB/s eta 0:00:21\n",
      "     --- ---------------------------------- 45.7/567.8 MB 26.2 MB/s eta 0:00:20\n",
      "     --- ---------------------------------- 47.0/567.8 MB 28.4 MB/s eta 0:00:19\n",
      "     --- ---------------------------------- 48.3/567.8 MB 27.3 MB/s eta 0:00:20\n",
      "     --- ---------------------------------- 49.3/567.8 MB 28.5 MB/s eta 0:00:19\n",
      "     --- ---------------------------------- 50.8/567.8 MB 28.5 MB/s eta 0:00:19\n",
      "     --- ---------------------------------- 52.2/567.8 MB 28.5 MB/s eta 0:00:19\n",
      "     --- ---------------------------------- 53.5/567.8 MB 28.5 MB/s eta 0:00:19\n",
      "     --- ---------------------------------- 54.7/567.8 MB 28.5 MB/s eta 0:00:19\n",
      "     --- ---------------------------------- 56.0/567.8 MB 28.4 MB/s eta 0:00:19\n",
      "     --- ---------------------------------- 56.4/567.8 MB 28.5 MB/s eta 0:00:18\n",
      "     --- ---------------------------------- 58.5/567.8 MB 28.4 MB/s eta 0:00:18\n",
      "     ---- --------------------------------- 60.1/567.8 MB 28.5 MB/s eta 0:00:18\n",
      "     ---- --------------------------------- 61.5/567.8 MB 29.8 MB/s eta 0:00:17\n",
      "     ---- --------------------------------- 63.1/567.8 MB 29.7 MB/s eta 0:00:17\n",
      "     ---- --------------------------------- 64.4/567.8 MB 29.7 MB/s eta 0:00:17\n",
      "     ---- --------------------------------- 65.8/567.8 MB 29.7 MB/s eta 0:00:17\n",
      "     ---- --------------------------------- 66.9/567.8 MB 32.7 MB/s eta 0:00:16\n",
      "     ---- --------------------------------- 67.7/567.8 MB 28.4 MB/s eta 0:00:18\n",
      "     ---- --------------------------------- 68.6/567.8 MB 28.5 MB/s eta 0:00:18\n",
      "     ---- --------------------------------- 69.7/567.8 MB 26.2 MB/s eta 0:00:19\n",
      "     ---- --------------------------------- 70.5/567.8 MB 25.2 MB/s eta 0:00:20\n",
      "     ---- --------------------------------- 71.7/567.8 MB 24.2 MB/s eta 0:00:21\n",
      "     ---- --------------------------------- 72.8/567.8 MB 23.4 MB/s eta 0:00:22\n",
      "     ---- --------------------------------- 74.0/567.8 MB 24.2 MB/s eta 0:00:21\n",
      "     ----- -------------------------------- 75.1/567.8 MB 23.4 MB/s eta 0:00:22\n",
      "     ----- -------------------------------- 76.4/567.8 MB 23.4 MB/s eta 0:00:21\n",
      "     ----- -------------------------------- 77.7/567.8 MB 24.2 MB/s eta 0:00:21\n",
      "     ----- -------------------------------- 78.7/567.8 MB 24.2 MB/s eta 0:00:21\n",
      "     ----- -------------------------------- 79.8/567.8 MB 25.1 MB/s eta 0:00:20\n",
      "     ----- -------------------------------- 81.3/567.8 MB 26.2 MB/s eta 0:00:19\n",
      "     ----- -------------------------------- 82.3/567.8 MB 26.2 MB/s eta 0:00:19\n",
      "     ----- -------------------------------- 83.5/567.8 MB 26.2 MB/s eta 0:00:19\n",
      "     ----- -------------------------------- 84.9/567.8 MB 27.3 MB/s eta 0:00:18\n",
      "     ----- -------------------------------- 85.9/567.8 MB 26.2 MB/s eta 0:00:19\n",
      "     ----- -------------------------------- 87.0/567.8 MB 25.2 MB/s eta 0:00:20\n",
      "     ----- -------------------------------- 88.2/567.8 MB 25.2 MB/s eta 0:00:20\n",
      "     ----- -------------------------------- 89.5/567.8 MB 25.1 MB/s eta 0:00:20\n",
      "     ------ ------------------------------- 90.8/567.8 MB 26.2 MB/s eta 0:00:19\n",
      "     ------ ------------------------------- 92.3/567.8 MB 27.3 MB/s eta 0:00:18\n",
      "     ------ ------------------------------- 93.8/567.8 MB 28.5 MB/s eta 0:00:17\n",
      "     ------ ------------------------------- 95.1/567.8 MB 28.5 MB/s eta 0:00:17\n",
      "     ------ ------------------------------- 96.6/567.8 MB 28.5 MB/s eta 0:00:17\n",
      "     ------ ------------------------------- 98.2/567.8 MB 29.7 MB/s eta 0:00:16\n",
      "     ------ ------------------------------- 99.4/567.8 MB 29.7 MB/s eta 0:00:16\n",
      "     ------ ------------------------------ 100.4/567.8 MB 29.7 MB/s eta 0:00:16\n",
      "     ------ ------------------------------ 101.9/567.8 MB 31.2 MB/s eta 0:00:15\n",
      "     ------ ------------------------------ 103.4/567.8 MB 29.8 MB/s eta 0:00:16\n",
      "     ------ ------------------------------ 104.8/567.8 MB 29.8 MB/s eta 0:00:16\n",
      "     ------ ------------------------------ 106.3/567.8 MB 29.7 MB/s eta 0:00:16\n",
      "     ------- ----------------------------- 107.9/567.8 MB 29.7 MB/s eta 0:00:16\n",
      "     ------- ----------------------------- 109.1/567.8 MB 29.7 MB/s eta 0:00:16\n",
      "     ------- ----------------------------- 110.5/567.8 MB 29.7 MB/s eta 0:00:16\n",
      "     ------- ----------------------------- 111.7/567.8 MB 29.7 MB/s eta 0:00:16\n",
      "     ------- ----------------------------- 113.2/567.8 MB 29.7 MB/s eta 0:00:16\n",
      "     ------- ----------------------------- 114.5/567.8 MB 28.5 MB/s eta 0:00:16\n",
      "     ------- ----------------------------- 115.9/567.8 MB 28.5 MB/s eta 0:00:16\n",
      "     ------- ----------------------------- 116.8/567.8 MB 27.3 MB/s eta 0:00:17\n",
      "     ------- ----------------------------- 118.1/567.8 MB 27.3 MB/s eta 0:00:17\n",
      "     ------- ----------------------------- 119.2/567.8 MB 27.3 MB/s eta 0:00:17\n",
      "     ------- ----------------------------- 120.5/567.8 MB 26.2 MB/s eta 0:00:18\n",
      "     ------- ----------------------------- 121.6/567.8 MB 27.3 MB/s eta 0:00:17\n",
      "     -------- ---------------------------- 122.9/567.8 MB 26.2 MB/s eta 0:00:18\n",
      "     -------- ---------------------------- 124.2/567.8 MB 25.2 MB/s eta 0:00:18\n",
      "     -------- ---------------------------- 124.9/567.8 MB 25.1 MB/s eta 0:00:18\n",
      "     -------- ---------------------------- 125.9/567.8 MB 25.2 MB/s eta 0:00:18\n",
      "     -------- ---------------------------- 127.2/567.8 MB 25.1 MB/s eta 0:00:18\n",
      "     -------- ---------------------------- 127.9/567.8 MB 22.5 MB/s eta 0:00:20\n",
      "     -------- ---------------------------- 128.4/567.8 MB 21.1 MB/s eta 0:00:21\n",
      "     -------- ---------------------------- 128.9/567.8 MB 20.5 MB/s eta 0:00:22\n",
      "     -------- ---------------------------- 129.4/567.8 MB 19.3 MB/s eta 0:00:23\n",
      "     -------- ---------------------------- 129.9/567.8 MB 18.2 MB/s eta 0:00:25\n",
      "     -------- ---------------------------- 130.4/567.8 MB 17.2 MB/s eta 0:00:26\n",
      "     -------- ---------------------------- 130.9/567.8 MB 16.4 MB/s eta 0:00:27\n",
      "     -------- ---------------------------- 131.4/567.8 MB 16.4 MB/s eta 0:00:27\n",
      "     -------- ---------------------------- 132.0/567.8 MB 15.6 MB/s eta 0:00:28\n",
      "     -------- ---------------------------- 132.6/567.8 MB 14.6 MB/s eta 0:00:30\n",
      "     -------- ---------------------------- 133.2/567.8 MB 14.2 MB/s eta 0:00:31\n",
      "     -------- ---------------------------- 133.8/567.8 MB 13.9 MB/s eta 0:00:32\n",
      "     -------- ---------------------------- 134.4/567.8 MB 13.4 MB/s eta 0:00:33\n",
      "     -------- ---------------------------- 135.0/567.8 MB 12.8 MB/s eta 0:00:34\n",
      "     -------- ---------------------------- 135.5/567.8 MB 12.8 MB/s eta 0:00:34\n",
      "     -------- ---------------------------- 136.0/567.8 MB 12.1 MB/s eta 0:00:36\n",
      "     -------- ---------------------------- 136.4/567.8 MB 12.4 MB/s eta 0:00:35\n",
      "     -------- ---------------------------- 136.8/567.8 MB 11.5 MB/s eta 0:00:38\n",
      "     -------- ---------------------------- 137.3/567.8 MB 11.3 MB/s eta 0:00:39\n",
      "     -------- ---------------------------- 137.7/567.8 MB 11.1 MB/s eta 0:00:39\n",
      "     -------- ---------------------------- 138.1/567.8 MB 10.7 MB/s eta 0:00:41\n",
      "     --------- --------------------------- 138.5/567.8 MB 10.7 MB/s eta 0:00:40\n",
      "     --------- --------------------------- 138.9/567.8 MB 10.6 MB/s eta 0:00:41\n",
      "     --------- --------------------------- 139.4/567.8 MB 10.6 MB/s eta 0:00:41\n",
      "     --------- --------------------------- 139.8/567.8 MB 10.6 MB/s eta 0:00:41\n",
      "     --------- --------------------------- 140.2/567.8 MB 10.4 MB/s eta 0:00:42\n",
      "     --------- --------------------------- 140.6/567.8 MB 10.4 MB/s eta 0:00:42\n",
      "     --------- --------------------------- 141.1/567.8 MB 10.2 MB/s eta 0:00:42\n",
      "     --------- --------------------------- 141.5/567.8 MB 10.1 MB/s eta 0:00:43\n",
      "     --------- --------------------------- 142.0/567.8 MB 10.1 MB/s eta 0:00:43\n",
      "     --------- ---------------------------- 142.4/567.8 MB 9.8 MB/s eta 0:00:44\n",
      "     --------- ---------------------------- 142.8/567.8 MB 9.8 MB/s eta 0:00:44\n",
      "     --------- ---------------------------- 143.4/567.8 MB 9.8 MB/s eta 0:00:44\n",
      "     --------- ---------------------------- 143.9/567.8 MB 9.6 MB/s eta 0:00:45\n",
      "     --------- ---------------------------- 144.5/567.8 MB 9.4 MB/s eta 0:00:46\n",
      "     --------- ---------------------------- 145.0/567.8 MB 9.2 MB/s eta 0:00:46\n",
      "     --------- ---------------------------- 145.8/567.8 MB 9.2 MB/s eta 0:00:46\n",
      "     --------- ---------------------------- 146.4/567.8 MB 9.2 MB/s eta 0:00:46\n",
      "     --------- ---------------------------- 146.9/567.8 MB 9.2 MB/s eta 0:00:46\n",
      "     --------- ---------------------------- 147.4/567.8 MB 9.2 MB/s eta 0:00:46\n",
      "     --------- ---------------------------- 147.8/567.8 MB 9.2 MB/s eta 0:00:46\n",
      "     --------- ---------------------------- 148.2/567.8 MB 9.2 MB/s eta 0:00:46\n",
      "     --------- ---------------------------- 148.7/567.8 MB 9.2 MB/s eta 0:00:46\n",
      "     --------- ---------------------------- 149.1/567.8 MB 9.4 MB/s eta 0:00:45\n",
      "     ---------- --------------------------- 149.5/567.8 MB 9.2 MB/s eta 0:00:46\n",
      "     ---------- --------------------------- 149.7/567.8 MB 9.2 MB/s eta 0:00:46\n",
      "     ---------- --------------------------- 150.0/567.8 MB 9.1 MB/s eta 0:00:46\n",
      "     ---------- --------------------------- 150.2/567.8 MB 9.0 MB/s eta 0:00:47\n",
      "     ---------- --------------------------- 150.4/567.8 MB 8.7 MB/s eta 0:00:48\n",
      "     ---------- --------------------------- 150.6/567.8 MB 8.6 MB/s eta 0:00:49\n",
      "     ---------- --------------------------- 150.9/567.8 MB 8.4 MB/s eta 0:00:50\n",
      "     ---------- --------------------------- 151.2/567.8 MB 8.3 MB/s eta 0:00:51\n",
      "     ---------- --------------------------- 151.4/567.8 MB 8.1 MB/s eta 0:00:52\n",
      "     ---------- --------------------------- 151.7/567.8 MB 8.0 MB/s eta 0:00:53\n",
      "     ---------- --------------------------- 151.9/567.8 MB 7.9 MB/s eta 0:00:53\n",
      "     ---------- --------------------------- 152.3/567.8 MB 7.8 MB/s eta 0:00:54\n",
      "     ---------- --------------------------- 152.4/567.8 MB 7.7 MB/s eta 0:00:54\n",
      "     ---------- --------------------------- 152.7/567.8 MB 7.6 MB/s eta 0:00:55\n",
      "     ---------- --------------------------- 153.0/567.8 MB 7.5 MB/s eta 0:00:56\n",
      "     ---------- --------------------------- 153.2/567.8 MB 7.4 MB/s eta 0:00:57\n",
      "     ---------- --------------------------- 153.5/567.8 MB 7.3 MB/s eta 0:00:57\n",
      "     ---------- --------------------------- 153.9/567.8 MB 7.2 MB/s eta 0:00:58\n",
      "     ---------- --------------------------- 154.3/567.8 MB 7.4 MB/s eta 0:00:57\n",
      "     ---------- --------------------------- 154.6/567.8 MB 7.2 MB/s eta 0:00:58\n",
      "     ---------- --------------------------- 155.0/567.8 MB 7.2 MB/s eta 0:00:58\n",
      "     ---------- --------------------------- 155.2/567.8 MB 7.1 MB/s eta 0:00:58\n",
      "     ---------- --------------------------- 155.6/567.8 MB 7.0 MB/s eta 0:01:00\n",
      "     ---------- --------------------------- 155.8/567.8 MB 6.9 MB/s eta 0:01:00\n",
      "     ---------- --------------------------- 156.1/567.8 MB 6.9 MB/s eta 0:01:00\n",
      "     ---------- --------------------------- 156.4/567.8 MB 6.7 MB/s eta 0:01:01\n",
      "     ---------- --------------------------- 156.7/567.8 MB 6.7 MB/s eta 0:01:02\n",
      "     ---------- --------------------------- 157.0/567.8 MB 6.6 MB/s eta 0:01:03\n",
      "     ---------- --------------------------- 157.4/567.8 MB 6.6 MB/s eta 0:01:03\n",
      "     ---------- --------------------------- 157.8/567.8 MB 6.6 MB/s eta 0:01:03\n",
      "     ---------- --------------------------- 158.2/567.8 MB 6.5 MB/s eta 0:01:03\n",
      "     ---------- --------------------------- 158.7/567.8 MB 6.5 MB/s eta 0:01:03\n",
      "     ---------- --------------------------- 159.1/567.8 MB 6.6 MB/s eta 0:01:02\n",
      "     ---------- --------------------------- 159.5/567.8 MB 6.5 MB/s eta 0:01:03\n",
      "     ---------- --------------------------- 159.9/567.8 MB 6.5 MB/s eta 0:01:03\n",
      "     ---------- --------------------------- 160.2/567.8 MB 6.6 MB/s eta 0:01:02\n",
      "     ---------- --------------------------- 160.6/567.8 MB 6.8 MB/s eta 0:01:01\n",
      "     ---------- --------------------------- 160.8/567.8 MB 6.8 MB/s eta 0:01:01\n",
      "     ---------- --------------------------- 161.2/567.8 MB 6.9 MB/s eta 0:00:59\n",
      "     ---------- --------------------------- 161.6/567.8 MB 7.0 MB/s eta 0:00:58\n",
      "     ---------- --------------------------- 162.0/567.8 MB 7.2 MB/s eta 0:00:57\n",
      "     ---------- --------------------------- 162.4/567.8 MB 7.4 MB/s eta 0:00:56\n",
      "     ---------- --------------------------- 162.8/567.8 MB 7.5 MB/s eta 0:00:54\n",
      "     ---------- --------------------------- 163.2/567.8 MB 7.6 MB/s eta 0:00:54\n",
      "     ---------- --------------------------- 163.6/567.8 MB 7.8 MB/s eta 0:00:52\n",
      "     ---------- --------------------------- 164.0/567.8 MB 7.9 MB/s eta 0:00:52\n",
      "     ---------- --------------------------- 164.3/567.8 MB 7.7 MB/s eta 0:00:53\n",
      "     ----------- -------------------------- 164.7/567.8 MB 7.8 MB/s eta 0:00:52\n",
      "     ----------- -------------------------- 165.0/567.8 MB 7.8 MB/s eta 0:00:52\n",
      "     ----------- -------------------------- 165.3/567.8 MB 7.9 MB/s eta 0:00:52\n",
      "     ----------- -------------------------- 165.7/567.8 MB 8.0 MB/s eta 0:00:51\n",
      "     ----------- -------------------------- 166.1/567.8 MB 8.0 MB/s eta 0:00:51\n",
      "     ----------- -------------------------- 166.5/567.8 MB 8.1 MB/s eta 0:00:50\n",
      "     ----------- -------------------------- 166.9/567.8 MB 8.2 MB/s eta 0:00:49\n",
      "     ----------- -------------------------- 167.3/567.8 MB 8.3 MB/s eta 0:00:49\n",
      "     ----------- -------------------------- 167.7/567.8 MB 8.3 MB/s eta 0:00:49\n",
      "     ----------- -------------------------- 168.2/567.8 MB 8.3 MB/s eta 0:00:49\n",
      "     ----------- -------------------------- 168.5/567.8 MB 8.3 MB/s eta 0:00:49\n",
      "     ----------- -------------------------- 168.8/567.8 MB 8.1 MB/s eta 0:00:50\n",
      "     ----------- -------------------------- 169.2/567.8 MB 8.3 MB/s eta 0:00:49\n",
      "     ----------- -------------------------- 169.6/567.8 MB 8.2 MB/s eta 0:00:49\n",
      "     ----------- -------------------------- 170.1/567.8 MB 8.2 MB/s eta 0:00:49\n",
      "     ----------- -------------------------- 170.6/567.8 MB 8.3 MB/s eta 0:00:48\n",
      "     ----------- -------------------------- 171.0/567.8 MB 8.6 MB/s eta 0:00:47\n",
      "     ----------- -------------------------- 171.6/567.8 MB 8.6 MB/s eta 0:00:46\n",
      "     ----------- -------------------------- 172.0/567.8 MB 8.7 MB/s eta 0:00:46\n",
      "     ----------- -------------------------- 172.5/567.8 MB 8.8 MB/s eta 0:00:45\n",
      "     ----------- -------------------------- 173.0/567.8 MB 9.0 MB/s eta 0:00:44\n",
      "     ----------- -------------------------- 173.5/567.8 MB 9.0 MB/s eta 0:00:44\n",
      "     ----------- -------------------------- 173.9/567.8 MB 9.2 MB/s eta 0:00:43\n",
      "     ----------- -------------------------- 174.5/567.8 MB 9.2 MB/s eta 0:00:43\n",
      "     ----------- -------------------------- 175.0/567.8 MB 9.4 MB/s eta 0:00:43\n",
      "     ----------- -------------------------- 175.4/567.8 MB 9.5 MB/s eta 0:00:42\n",
      "     ----------- -------------------------- 175.8/567.8 MB 9.5 MB/s eta 0:00:42\n",
      "     ----------- -------------------------- 176.4/567.8 MB 9.8 MB/s eta 0:00:41\n",
      "     ----------- -------------------------- 176.9/567.8 MB 9.9 MB/s eta 0:00:40\n",
      "     ----------- -------------------------- 177.4/567.8 MB 9.9 MB/s eta 0:00:40\n",
      "     ----------- -------------------------- 177.8/567.8 MB 9.9 MB/s eta 0:00:40\n",
      "     ----------- -------------------------- 178.3/567.8 MB 9.9 MB/s eta 0:00:40\n",
      "     ----------- ------------------------- 178.7/567.8 MB 10.1 MB/s eta 0:00:39\n",
      "     ----------- ------------------------- 179.3/567.8 MB 10.2 MB/s eta 0:00:38\n",
      "     ----------- ------------------------- 179.8/567.8 MB 10.4 MB/s eta 0:00:38\n",
      "     ----------- ------------------------- 180.3/567.8 MB 10.6 MB/s eta 0:00:37\n",
      "     ----------- ------------------------- 180.8/567.8 MB 10.6 MB/s eta 0:00:37\n",
      "     ----------- ------------------------- 181.4/567.8 MB 10.6 MB/s eta 0:00:37\n",
      "     ----------- ------------------------- 181.9/567.8 MB 10.7 MB/s eta 0:00:36\n",
      "     ----------- ------------------------- 182.5/567.8 MB 10.7 MB/s eta 0:00:36\n",
      "     ----------- ------------------------- 183.0/567.8 MB 10.7 MB/s eta 0:00:36\n",
      "     ----------- ------------------------- 183.3/567.8 MB 10.6 MB/s eta 0:00:37\n",
      "     ----------- ------------------------- 183.9/567.8 MB 10.6 MB/s eta 0:00:37\n",
      "     ------------ ------------------------ 184.5/567.8 MB 10.7 MB/s eta 0:00:36\n",
      "     ------------ ------------------------ 185.0/567.8 MB 10.7 MB/s eta 0:00:36\n",
      "     ------------ ------------------------ 185.6/567.8 MB 10.9 MB/s eta 0:00:36\n",
      "     ------------ ------------------------ 186.2/567.8 MB 11.1 MB/s eta 0:00:35\n",
      "     ------------ ------------------------ 186.8/567.8 MB 11.1 MB/s eta 0:00:35\n",
      "     ------------ ------------------------ 187.3/567.8 MB 11.1 MB/s eta 0:00:35\n",
      "     ------------ ------------------------ 187.9/567.8 MB 11.3 MB/s eta 0:00:34\n",
      "     ------------ ------------------------ 188.4/567.8 MB 11.3 MB/s eta 0:00:34\n",
      "     ------------ ------------------------ 188.8/567.8 MB 11.5 MB/s eta 0:00:34\n",
      "     ------------ ------------------------ 189.4/567.8 MB 11.3 MB/s eta 0:00:34\n",
      "     ------------ ------------------------ 190.0/567.8 MB 11.5 MB/s eta 0:00:33\n",
      "     ------------ ------------------------ 190.5/567.8 MB 11.5 MB/s eta 0:00:33\n",
      "     ------------ ------------------------ 191.1/567.8 MB 11.9 MB/s eta 0:00:32\n",
      "     ------------ ------------------------ 191.7/567.8 MB 11.7 MB/s eta 0:00:33\n",
      "     ------------ ------------------------ 192.3/567.8 MB 11.9 MB/s eta 0:00:32\n",
      "     ------------ ------------------------ 192.8/567.8 MB 11.7 MB/s eta 0:00:33\n",
      "     ------------ ------------------------ 193.3/567.8 MB 11.9 MB/s eta 0:00:32\n",
      "     ------------ ------------------------ 194.1/567.8 MB 12.1 MB/s eta 0:00:31\n",
      "     ------------ ------------------------ 194.7/567.8 MB 12.4 MB/s eta 0:00:31\n",
      "     ------------ ------------------------ 195.3/567.8 MB 12.4 MB/s eta 0:00:31\n",
      "     ------------ ------------------------ 195.9/567.8 MB 12.4 MB/s eta 0:00:31\n",
      "     ------------ ------------------------ 196.4/567.8 MB 12.6 MB/s eta 0:00:30\n",
      "     ------------ ------------------------ 197.0/567.8 MB 12.4 MB/s eta 0:00:31\n",
      "     ------------ ------------------------ 197.6/567.8 MB 12.4 MB/s eta 0:00:30\n",
      "     ------------ ------------------------ 198.2/567.8 MB 12.4 MB/s eta 0:00:30\n",
      "     ------------ ------------------------ 198.8/567.8 MB 12.6 MB/s eta 0:00:30\n",
      "     ------------ ------------------------ 199.4/567.8 MB 12.9 MB/s eta 0:00:29\n",
      "     ------------- ----------------------- 200.0/567.8 MB 12.8 MB/s eta 0:00:29\n",
      "     ------------- ----------------------- 200.6/567.8 MB 12.8 MB/s eta 0:00:29\n",
      "     ------------- ----------------------- 201.2/567.8 MB 12.8 MB/s eta 0:00:29\n",
      "     ------------- ----------------------- 201.9/567.8 MB 13.1 MB/s eta 0:00:28\n",
      "     ------------- ----------------------- 202.4/567.8 MB 13.1 MB/s eta 0:00:28\n",
      "     ------------- ----------------------- 202.9/567.8 MB 13.1 MB/s eta 0:00:28\n",
      "     ------------- ----------------------- 203.4/567.8 MB 12.8 MB/s eta 0:00:29\n",
      "     ------------- ----------------------- 204.1/567.8 MB 12.8 MB/s eta 0:00:29\n",
      "     ------------- ----------------------- 204.9/567.8 MB 13.1 MB/s eta 0:00:28\n",
      "     ------------- ----------------------- 205.5/567.8 MB 13.1 MB/s eta 0:00:28\n",
      "     ------------- ----------------------- 206.1/567.8 MB 13.1 MB/s eta 0:00:28\n",
      "     ------------- ----------------------- 206.8/567.8 MB 13.4 MB/s eta 0:00:27\n",
      "     ------------- ----------------------- 207.4/567.8 MB 13.4 MB/s eta 0:00:27\n",
      "     ------------- ----------------------- 207.9/567.8 MB 13.4 MB/s eta 0:00:27\n",
      "     ------------- ----------------------- 208.6/567.8 MB 13.4 MB/s eta 0:00:27\n",
      "     ------------- ----------------------- 209.2/567.8 MB 13.6 MB/s eta 0:00:27\n",
      "     ------------- ----------------------- 210.0/567.8 MB 13.9 MB/s eta 0:00:26\n",
      "     ------------- ----------------------- 210.6/567.8 MB 13.9 MB/s eta 0:00:26\n",
      "     ------------- ----------------------- 211.5/567.8 MB 14.2 MB/s eta 0:00:26\n",
      "     ------------- ----------------------- 211.8/567.8 MB 13.9 MB/s eta 0:00:26\n",
      "     ------------- ----------------------- 212.5/567.8 MB 13.9 MB/s eta 0:00:26\n",
      "     ------------- ----------------------- 213.0/567.8 MB 13.9 MB/s eta 0:00:26\n",
      "     ------------- ----------------------- 213.5/567.8 MB 13.6 MB/s eta 0:00:26\n",
      "     ------------- ----------------------- 214.0/567.8 MB 13.6 MB/s eta 0:00:26\n",
      "     ------------- ----------------------- 214.4/567.8 MB 13.4 MB/s eta 0:00:27\n",
      "     -------------- ---------------------- 214.9/567.8 MB 13.4 MB/s eta 0:00:27\n",
      "     -------------- ---------------------- 215.3/567.8 MB 12.8 MB/s eta 0:00:28\n",
      "     -------------- ---------------------- 215.7/567.8 MB 12.6 MB/s eta 0:00:28\n",
      "     -------------- ---------------------- 216.3/567.8 MB 12.6 MB/s eta 0:00:28\n",
      "     -------------- ---------------------- 216.8/567.8 MB 12.6 MB/s eta 0:00:28\n",
      "     -------------- ---------------------- 217.4/567.8 MB 12.6 MB/s eta 0:00:28\n",
      "     -------------- ---------------------- 218.0/567.8 MB 12.3 MB/s eta 0:00:29\n",
      "     -------------- ---------------------- 218.5/567.8 MB 12.1 MB/s eta 0:00:29\n",
      "     -------------- ---------------------- 219.1/567.8 MB 11.9 MB/s eta 0:00:30\n",
      "     -------------- ---------------------- 219.6/567.8 MB 11.9 MB/s eta 0:00:30\n",
      "     -------------- ---------------------- 220.1/567.8 MB 11.7 MB/s eta 0:00:30\n",
      "     -------------- ---------------------- 220.7/567.8 MB 11.5 MB/s eta 0:00:31\n",
      "     -------------- ---------------------- 221.3/567.8 MB 11.7 MB/s eta 0:00:30\n",
      "     -------------- ---------------------- 221.9/567.8 MB 11.5 MB/s eta 0:00:31\n",
      "     -------------- ---------------------- 222.5/567.8 MB 11.5 MB/s eta 0:00:31\n",
      "     -------------- ---------------------- 223.0/567.8 MB 11.5 MB/s eta 0:00:31\n",
      "     -------------- ---------------------- 223.5/567.8 MB 11.5 MB/s eta 0:00:30\n",
      "     -------------- ---------------------- 224.2/567.8 MB 11.9 MB/s eta 0:00:29\n",
      "     -------------- ---------------------- 224.8/567.8 MB 11.9 MB/s eta 0:00:29\n",
      "     -------------- ---------------------- 225.5/567.8 MB 12.1 MB/s eta 0:00:29\n",
      "     -------------- ---------------------- 226.1/567.8 MB 12.4 MB/s eta 0:00:28\n",
      "     -------------- ---------------------- 226.8/567.8 MB 12.4 MB/s eta 0:00:28\n",
      "     -------------- ---------------------- 227.4/567.8 MB 12.3 MB/s eta 0:00:28\n",
      "     -------------- ---------------------- 228.0/567.8 MB 12.6 MB/s eta 0:00:28\n",
      "     -------------- ---------------------- 228.5/567.8 MB 12.4 MB/s eta 0:00:28\n",
      "     -------------- ---------------------- 229.1/567.8 MB 12.8 MB/s eta 0:00:27\n",
      "     -------------- ---------------------- 229.6/567.8 MB 12.9 MB/s eta 0:00:27\n",
      "     -------------- ---------------------- 230.2/567.8 MB 12.8 MB/s eta 0:00:27\n",
      "     --------------- --------------------- 230.8/567.8 MB 12.8 MB/s eta 0:00:27\n",
      "     --------------- --------------------- 231.4/567.8 MB 12.8 MB/s eta 0:00:27\n",
      "     --------------- --------------------- 232.0/567.8 MB 12.8 MB/s eta 0:00:27\n",
      "     --------------- --------------------- 232.6/567.8 MB 12.8 MB/s eta 0:00:27\n",
      "     --------------- --------------------- 233.2/567.8 MB 13.1 MB/s eta 0:00:26\n",
      "     --------------- --------------------- 233.9/567.8 MB 13.1 MB/s eta 0:00:26\n",
      "     --------------- --------------------- 234.5/567.8 MB 13.1 MB/s eta 0:00:26\n",
      "     --------------- --------------------- 235.1/567.8 MB 13.1 MB/s eta 0:00:26\n",
      "     --------------- --------------------- 235.8/567.8 MB 13.1 MB/s eta 0:00:26\n",
      "     --------------- --------------------- 236.4/567.8 MB 13.1 MB/s eta 0:00:26\n",
      "     --------------- --------------------- 236.9/567.8 MB 13.1 MB/s eta 0:00:26\n",
      "     --------------- --------------------- 237.6/567.8 MB 12.8 MB/s eta 0:00:26\n",
      "     --------------- --------------------- 238.3/567.8 MB 13.1 MB/s eta 0:00:26\n",
      "     --------------- --------------------- 238.9/567.8 MB 13.1 MB/s eta 0:00:26\n",
      "     --------------- --------------------- 239.5/567.8 MB 13.1 MB/s eta 0:00:26\n",
      "     --------------- --------------------- 240.0/567.8 MB 13.1 MB/s eta 0:00:26\n",
      "     --------------- --------------------- 240.4/567.8 MB 13.1 MB/s eta 0:00:25\n",
      "     --------------- --------------------- 240.9/567.8 MB 12.8 MB/s eta 0:00:26\n",
      "     --------------- --------------------- 241.3/567.8 MB 12.6 MB/s eta 0:00:26\n",
      "     --------------- --------------------- 241.8/567.8 MB 12.4 MB/s eta 0:00:27\n",
      "     --------------- --------------------- 242.3/567.8 MB 12.4 MB/s eta 0:00:27\n",
      "     --------------- --------------------- 242.8/567.8 MB 12.1 MB/s eta 0:00:27\n",
      "     --------------- --------------------- 243.2/567.8 MB 11.9 MB/s eta 0:00:28\n",
      "     --------------- --------------------- 243.7/567.8 MB 11.7 MB/s eta 0:00:28\n",
      "     --------------- --------------------- 244.2/567.8 MB 11.7 MB/s eta 0:00:28\n",
      "     --------------- --------------------- 244.6/567.8 MB 11.5 MB/s eta 0:00:29\n",
      "     --------------- --------------------- 245.1/567.8 MB 11.3 MB/s eta 0:00:29\n",
      "     ---------------- -------------------- 245.6/567.8 MB 11.3 MB/s eta 0:00:29\n",
      "     ---------------- -------------------- 246.0/567.8 MB 11.1 MB/s eta 0:00:30\n",
      "     ---------------- -------------------- 246.5/567.8 MB 11.1 MB/s eta 0:00:29\n",
      "     ---------------- -------------------- 247.0/567.8 MB 10.9 MB/s eta 0:00:30\n",
      "     ---------------- -------------------- 247.5/567.8 MB 10.9 MB/s eta 0:00:30\n",
      "     ---------------- -------------------- 248.0/567.8 MB 10.7 MB/s eta 0:00:30\n",
      "     ---------------- -------------------- 248.5/567.8 MB 10.6 MB/s eta 0:00:31\n",
      "     ---------------- -------------------- 249.0/567.8 MB 10.4 MB/s eta 0:00:31\n",
      "     ---------------- -------------------- 249.5/567.8 MB 10.4 MB/s eta 0:00:31\n",
      "     ---------------- -------------------- 250.1/567.8 MB 10.4 MB/s eta 0:00:31\n",
      "     ---------------- -------------------- 250.5/567.8 MB 10.4 MB/s eta 0:00:31\n",
      "     ---------------- -------------------- 251.1/567.8 MB 10.4 MB/s eta 0:00:31\n",
      "     ---------------- -------------------- 251.6/567.8 MB 10.6 MB/s eta 0:00:30\n",
      "     ---------------- -------------------- 252.1/567.8 MB 10.6 MB/s eta 0:00:30\n",
      "     ---------------- -------------------- 252.6/567.8 MB 10.6 MB/s eta 0:00:30\n",
      "     ---------------- -------------------- 253.2/567.8 MB 10.7 MB/s eta 0:00:30\n",
      "     ---------------- -------------------- 253.7/567.8 MB 10.7 MB/s eta 0:00:30\n",
      "     ---------------- -------------------- 254.2/567.8 MB 10.9 MB/s eta 0:00:29\n",
      "     ---------------- -------------------- 254.8/567.8 MB 10.9 MB/s eta 0:00:29\n",
      "     ---------------- -------------------- 255.3/567.8 MB 11.3 MB/s eta 0:00:28\n",
      "     ---------------- -------------------- 256.0/567.8 MB 11.7 MB/s eta 0:00:27\n",
      "     ---------------- -------------------- 256.5/567.8 MB 11.7 MB/s eta 0:00:27\n",
      "     ---------------- -------------------- 257.1/567.8 MB 11.7 MB/s eta 0:00:27\n",
      "     ---------------- -------------------- 257.7/567.8 MB 11.7 MB/s eta 0:00:27\n",
      "     ---------------- -------------------- 258.2/567.8 MB 11.7 MB/s eta 0:00:27\n",
      "     ---------------- -------------------- 258.8/567.8 MB 11.9 MB/s eta 0:00:26\n",
      "     ---------------- -------------------- 259.3/567.8 MB 11.9 MB/s eta 0:00:26\n",
      "     ---------------- -------------------- 260.0/567.8 MB 11.9 MB/s eta 0:00:26\n",
      "     ---------------- -------------------- 260.5/567.8 MB 12.1 MB/s eta 0:00:26\n",
      "     ----------------- ------------------- 261.1/567.8 MB 12.1 MB/s eta 0:00:26\n",
      "     ----------------- ------------------- 261.7/567.8 MB 12.1 MB/s eta 0:00:26\n",
      "     ----------------- ------------------- 262.3/567.8 MB 12.1 MB/s eta 0:00:26\n",
      "     ----------------- ------------------- 262.9/567.8 MB 12.3 MB/s eta 0:00:25\n",
      "     ----------------- ------------------- 263.5/567.8 MB 12.4 MB/s eta 0:00:25\n",
      "     ----------------- ------------------- 264.1/567.8 MB 12.6 MB/s eta 0:00:25\n",
      "     ----------------- ------------------- 264.7/567.8 MB 12.6 MB/s eta 0:00:25\n",
      "     ----------------- ------------------- 265.4/567.8 MB 12.6 MB/s eta 0:00:25\n",
      "     ----------------- ------------------- 266.1/567.8 MB 12.8 MB/s eta 0:00:24\n",
      "     ----------------- ------------------- 266.7/567.8 MB 12.8 MB/s eta 0:00:24\n",
      "     ----------------- ------------------- 267.3/567.8 MB 13.1 MB/s eta 0:00:23\n",
      "     ----------------- ------------------- 268.0/567.8 MB 13.1 MB/s eta 0:00:23\n",
      "     ----------------- ------------------- 268.6/567.8 MB 13.1 MB/s eta 0:00:23\n",
      "     ----------------- ------------------- 269.2/567.8 MB 13.1 MB/s eta 0:00:23\n",
      "     ----------------- ------------------- 269.8/567.8 MB 13.6 MB/s eta 0:00:22\n",
      "     ----------------- ------------------- 270.5/567.8 MB 13.6 MB/s eta 0:00:22\n",
      "     ----------------- ------------------- 271.2/567.8 MB 13.6 MB/s eta 0:00:22\n",
      "     ----------------- ------------------- 271.8/567.8 MB 13.6 MB/s eta 0:00:22\n",
      "     ----------------- ------------------- 272.4/567.8 MB 13.9 MB/s eta 0:00:22\n",
      "     ----------------- ------------------- 273.1/567.8 MB 13.9 MB/s eta 0:00:22\n",
      "     ----------------- ------------------- 273.9/567.8 MB 14.2 MB/s eta 0:00:21\n",
      "     ----------------- ------------------- 274.6/567.8 MB 14.6 MB/s eta 0:00:21\n",
      "     ----------------- ------------------- 275.4/567.8 MB 14.5 MB/s eta 0:00:21\n",
      "     ----------------- ------------------- 276.1/567.8 MB 14.6 MB/s eta 0:00:21\n",
      "     ------------------ ------------------ 276.9/567.8 MB 14.9 MB/s eta 0:00:20\n",
      "     ------------------ ------------------ 277.6/567.8 MB 14.9 MB/s eta 0:00:20\n",
      "     ------------------ ------------------ 278.4/567.8 MB 14.9 MB/s eta 0:00:20\n",
      "     ------------------ ------------------ 279.2/567.8 MB 15.2 MB/s eta 0:00:19\n",
      "     ------------------ ------------------ 279.9/567.8 MB 15.6 MB/s eta 0:00:19\n",
      "     ------------------ ------------------ 280.6/567.8 MB 15.6 MB/s eta 0:00:19\n",
      "     ------------------ ------------------ 281.3/567.8 MB 15.6 MB/s eta 0:00:19\n",
      "     ------------------ ------------------ 282.0/567.8 MB 15.6 MB/s eta 0:00:19\n",
      "     ------------------ ------------------ 282.7/567.8 MB 15.6 MB/s eta 0:00:19\n",
      "     ------------------ ------------------ 283.4/567.8 MB 16.0 MB/s eta 0:00:18\n",
      "     ------------------ ------------------ 284.1/567.8 MB 15.6 MB/s eta 0:00:19\n",
      "     ------------------ ------------------ 284.9/567.8 MB 15.6 MB/s eta 0:00:19\n",
      "     ------------------ ------------------ 285.5/567.8 MB 15.6 MB/s eta 0:00:19\n",
      "     ------------------ ------------------ 286.3/567.8 MB 15.6 MB/s eta 0:00:19\n",
      "     ------------------ ------------------ 287.0/567.8 MB 15.6 MB/s eta 0:00:19\n",
      "     ------------------ ------------------ 287.7/567.8 MB 15.6 MB/s eta 0:00:18\n",
      "     ------------------ ------------------ 288.4/567.8 MB 15.2 MB/s eta 0:00:19\n",
      "     ------------------ ------------------ 289.2/567.8 MB 15.6 MB/s eta 0:00:18\n",
      "     ------------------ ------------------ 289.9/567.8 MB 15.2 MB/s eta 0:00:19\n",
      "     ------------------ ------------------ 290.7/567.8 MB 15.6 MB/s eta 0:00:18\n",
      "     ------------------ ------------------ 291.4/567.8 MB 15.6 MB/s eta 0:00:18\n",
      "     ------------------- ----------------- 292.2/567.8 MB 15.6 MB/s eta 0:00:18\n",
      "     ------------------- ----------------- 293.0/567.8 MB 16.0 MB/s eta 0:00:18\n",
      "     ------------------- ----------------- 293.8/567.8 MB 16.0 MB/s eta 0:00:18\n",
      "     ------------------- ----------------- 294.6/567.8 MB 16.0 MB/s eta 0:00:18\n",
      "     ------------------- ----------------- 295.3/567.8 MB 16.0 MB/s eta 0:00:18\n",
      "     ------------------- ----------------- 296.0/567.8 MB 16.0 MB/s eta 0:00:18\n",
      "     ------------------- ----------------- 296.9/567.8 MB 16.4 MB/s eta 0:00:17\n",
      "     ------------------- ----------------- 297.7/567.8 MB 16.4 MB/s eta 0:00:17\n",
      "     ------------------- ----------------- 298.5/567.8 MB 16.4 MB/s eta 0:00:17\n",
      "     ------------------- ----------------- 299.2/567.8 MB 16.8 MB/s eta 0:00:16\n",
      "     ------------------- ----------------- 299.9/567.8 MB 16.8 MB/s eta 0:00:16\n",
      "     ------------------- ----------------- 300.7/567.8 MB 16.8 MB/s eta 0:00:16\n",
      "     ------------------- ----------------- 301.5/567.8 MB 16.4 MB/s eta 0:00:17\n",
      "     ------------------- ----------------- 302.2/567.8 MB 16.4 MB/s eta 0:00:17\n",
      "     ------------------- ----------------- 303.0/567.8 MB 16.8 MB/s eta 0:00:16\n",
      "     ------------------- ----------------- 303.9/567.8 MB 16.4 MB/s eta 0:00:17\n",
      "     ------------------- ----------------- 304.7/567.8 MB 16.4 MB/s eta 0:00:17\n",
      "     ------------------- ----------------- 305.6/567.8 MB 16.8 MB/s eta 0:00:16\n",
      "     ------------------- ----------------- 306.4/567.8 MB 16.8 MB/s eta 0:00:16\n",
      "     -------------------- ---------------- 307.1/567.8 MB 17.2 MB/s eta 0:00:16\n",
      "     -------------------- ---------------- 308.0/567.8 MB 16.8 MB/s eta 0:00:16\n",
      "     -------------------- ---------------- 308.7/567.8 MB 16.8 MB/s eta 0:00:16\n",
      "     -------------------- ---------------- 309.5/567.8 MB 16.8 MB/s eta 0:00:16\n",
      "     -------------------- ---------------- 310.3/567.8 MB 17.2 MB/s eta 0:00:15\n",
      "     -------------------- ---------------- 311.1/567.8 MB 17.2 MB/s eta 0:00:15\n",
      "     -------------------- ---------------- 311.9/567.8 MB 17.2 MB/s eta 0:00:15\n",
      "     -------------------- ---------------- 312.7/567.8 MB 17.2 MB/s eta 0:00:15\n",
      "     -------------------- ---------------- 313.5/567.8 MB 17.7 MB/s eta 0:00:15\n",
      "     -------------------- ---------------- 314.4/567.8 MB 17.7 MB/s eta 0:00:15\n",
      "     -------------------- ---------------- 315.1/567.8 MB 17.3 MB/s eta 0:00:15\n",
      "     -------------------- ---------------- 315.9/567.8 MB 17.3 MB/s eta 0:00:15\n",
      "     -------------------- ---------------- 316.6/567.8 MB 17.2 MB/s eta 0:00:15\n",
      "     -------------------- ---------------- 317.4/567.8 MB 16.8 MB/s eta 0:00:15\n",
      "     -------------------- ---------------- 318.1/567.8 MB 16.8 MB/s eta 0:00:15\n",
      "     -------------------- ---------------- 319.0/567.8 MB 17.2 MB/s eta 0:00:15\n",
      "     -------------------- ---------------- 319.9/567.8 MB 17.2 MB/s eta 0:00:15\n",
      "     -------------------- ---------------- 320.7/567.8 MB 17.2 MB/s eta 0:00:15\n",
      "     -------------------- ---------------- 321.6/567.8 MB 17.3 MB/s eta 0:00:15\n",
      "     --------------------- --------------- 322.5/567.8 MB 17.3 MB/s eta 0:00:15\n",
      "     --------------------- --------------- 323.3/567.8 MB 17.7 MB/s eta 0:00:14\n",
      "     --------------------- --------------- 324.2/567.8 MB 17.7 MB/s eta 0:00:14\n",
      "     --------------------- --------------- 325.0/567.8 MB 17.7 MB/s eta 0:00:14\n",
      "     --------------------- --------------- 325.9/567.8 MB 18.2 MB/s eta 0:00:14\n",
      "     --------------------- --------------- 326.8/567.8 MB 18.2 MB/s eta 0:00:14\n",
      "     --------------------- --------------- 327.6/567.8 MB 18.2 MB/s eta 0:00:14\n",
      "     --------------------- --------------- 328.7/567.8 MB 18.7 MB/s eta 0:00:13\n",
      "     --------------------- --------------- 329.5/567.8 MB 18.7 MB/s eta 0:00:13\n",
      "     --------------------- --------------- 330.4/567.8 MB 18.7 MB/s eta 0:00:13\n",
      "     --------------------- --------------- 331.3/567.8 MB 18.7 MB/s eta 0:00:13\n",
      "     --------------------- --------------- 332.2/567.8 MB 18.7 MB/s eta 0:00:13\n",
      "     --------------------- --------------- 333.1/567.8 MB 19.3 MB/s eta 0:00:13\n",
      "     --------------------- --------------- 334.0/567.8 MB 19.2 MB/s eta 0:00:13\n",
      "     --------------------- --------------- 334.9/567.8 MB 19.2 MB/s eta 0:00:13\n",
      "     --------------------- --------------- 335.9/567.8 MB 19.3 MB/s eta 0:00:13\n",
      "     --------------------- --------------- 336.7/567.8 MB 19.3 MB/s eta 0:00:12\n",
      "     ---------------------- -------------- 337.7/567.8 MB 19.3 MB/s eta 0:00:12\n",
      "     ---------------------- -------------- 338.6/567.8 MB 19.3 MB/s eta 0:00:12\n",
      "     ---------------------- -------------- 339.6/567.8 MB 19.8 MB/s eta 0:00:12\n",
      "     ---------------------- -------------- 340.6/567.8 MB 19.8 MB/s eta 0:00:12\n",
      "     ---------------------- -------------- 341.5/567.8 MB 19.8 MB/s eta 0:00:12\n",
      "     ---------------------- -------------- 342.4/567.8 MB 19.8 MB/s eta 0:00:12\n",
      "     ---------------------- -------------- 343.4/567.8 MB 19.9 MB/s eta 0:00:12\n",
      "     ---------------------- -------------- 344.3/567.8 MB 19.9 MB/s eta 0:00:12\n",
      "     ---------------------- -------------- 345.4/567.8 MB 20.5 MB/s eta 0:00:11\n",
      "     ---------------------- -------------- 346.2/567.8 MB 20.5 MB/s eta 0:00:11\n",
      "     ---------------------- -------------- 347.2/567.8 MB 20.5 MB/s eta 0:00:11\n",
      "     ---------------------- -------------- 348.0/567.8 MB 19.8 MB/s eta 0:00:12\n",
      "     ---------------------- -------------- 348.8/567.8 MB 19.8 MB/s eta 0:00:12\n",
      "     ---------------------- -------------- 349.9/567.8 MB 19.8 MB/s eta 0:00:11\n",
      "     ---------------------- -------------- 350.8/567.8 MB 20.5 MB/s eta 0:00:11\n",
      "     ---------------------- -------------- 351.7/567.8 MB 20.5 MB/s eta 0:00:11\n",
      "     ---------------------- -------------- 352.7/567.8 MB 20.5 MB/s eta 0:00:11\n",
      "     ----------------------- ------------- 353.6/567.8 MB 20.5 MB/s eta 0:00:11\n",
      "     ----------------------- ------------- 354.5/567.8 MB 20.5 MB/s eta 0:00:11\n",
      "     ----------------------- ------------- 355.5/567.8 MB 20.5 MB/s eta 0:00:11\n",
      "     ----------------------- ------------- 356.6/567.8 MB 20.5 MB/s eta 0:00:11\n",
      "     ----------------------- ------------- 357.5/567.8 MB 20.5 MB/s eta 0:00:11\n",
      "     ----------------------- ------------- 358.4/567.8 MB 21.1 MB/s eta 0:00:10\n",
      "     ----------------------- ------------- 359.5/567.8 MB 21.1 MB/s eta 0:00:10\n",
      "     ----------------------- ------------- 360.6/567.8 MB 21.1 MB/s eta 0:00:10\n",
      "     ----------------------- ------------- 361.5/567.8 MB 21.1 MB/s eta 0:00:10\n",
      "     ----------------------- ------------- 362.5/567.8 MB 21.1 MB/s eta 0:00:10\n",
      "     ----------------------- ------------- 363.5/567.8 MB 21.1 MB/s eta 0:00:10\n",
      "     ----------------------- ------------- 364.5/567.8 MB 21.1 MB/s eta 0:00:10\n",
      "     ----------------------- ------------- 365.4/567.8 MB 21.1 MB/s eta 0:00:10\n",
      "     ----------------------- ------------- 366.3/567.8 MB 21.1 MB/s eta 0:00:10\n",
      "     ----------------------- ------------- 367.4/567.8 MB 21.1 MB/s eta 0:00:10\n",
      "     ------------------------ ------------ 368.4/567.8 MB 21.1 MB/s eta 0:00:10\n",
      "     ------------------------ ------------ 369.4/567.8 MB 21.1 MB/s eta 0:00:10\n",
      "     ------------------------ ------------ 370.4/567.8 MB 21.9 MB/s eta 0:00:10\n",
      "     ------------------------ ------------ 371.4/567.8 MB 21.1 MB/s eta 0:00:10\n",
      "     ------------------------ ------------ 372.5/567.8 MB 21.8 MB/s eta 0:00:09\n",
      "     ------------------------ ------------ 373.5/567.8 MB 21.8 MB/s eta 0:00:09\n",
      "     ------------------------ ------------ 374.5/567.8 MB 21.8 MB/s eta 0:00:09\n",
      "     ------------------------ ------------ 375.5/567.8 MB 21.8 MB/s eta 0:00:09\n",
      "     ------------------------ ------------ 376.5/567.8 MB 21.8 MB/s eta 0:00:09\n",
      "     ------------------------ ------------ 377.7/567.8 MB 22.6 MB/s eta 0:00:09\n",
      "     ------------------------ ------------ 378.7/567.8 MB 21.9 MB/s eta 0:00:09\n",
      "     ------------------------ ------------ 379.8/567.8 MB 22.6 MB/s eta 0:00:09\n",
      "     ------------------------ ------------ 380.8/567.8 MB 21.8 MB/s eta 0:00:09\n",
      "     ------------------------ ------------ 381.7/567.8 MB 21.8 MB/s eta 0:00:09\n",
      "     ------------------------ ------------ 382.7/567.8 MB 21.8 MB/s eta 0:00:09\n",
      "     ------------------------- ----------- 383.8/567.8 MB 21.8 MB/s eta 0:00:09\n",
      "     ------------------------- ----------- 384.8/567.8 MB 21.8 MB/s eta 0:00:09\n",
      "     ------------------------- ----------- 385.8/567.8 MB 21.9 MB/s eta 0:00:09\n",
      "     ------------------------- ----------- 386.7/567.8 MB 21.9 MB/s eta 0:00:09\n",
      "     ------------------------- ----------- 387.7/567.8 MB 21.8 MB/s eta 0:00:09\n",
      "     ------------------------- ----------- 388.7/567.8 MB 21.1 MB/s eta 0:00:09\n",
      "     ------------------------- ----------- 389.8/567.8 MB 21.1 MB/s eta 0:00:09\n",
      "     ------------------------- ----------- 390.7/567.8 MB 21.1 MB/s eta 0:00:09\n",
      "     ------------------------- ----------- 391.7/567.8 MB 21.1 MB/s eta 0:00:09\n",
      "     ------------------------- ----------- 392.6/567.8 MB 21.1 MB/s eta 0:00:09\n",
      "     ------------------------- ----------- 393.6/567.8 MB 21.1 MB/s eta 0:00:09\n",
      "     ------------------------- ----------- 394.5/567.8 MB 20.5 MB/s eta 0:00:09\n",
      "     ------------------------- ----------- 395.2/567.8 MB 19.9 MB/s eta 0:00:09\n",
      "     ------------------------- ----------- 396.0/567.8 MB 19.8 MB/s eta 0:00:09\n",
      "     ------------------------- ----------- 396.8/567.8 MB 19.3 MB/s eta 0:00:09\n",
      "     ------------------------- ----------- 397.6/567.8 MB 19.9 MB/s eta 0:00:09\n",
      "     ------------------------- ----------- 398.4/567.8 MB 18.7 MB/s eta 0:00:10\n",
      "     -------------------------- ---------- 399.2/567.8 MB 18.2 MB/s eta 0:00:10\n",
      "     -------------------------- ---------- 400.0/567.8 MB 18.7 MB/s eta 0:00:09\n",
      "     -------------------------- ---------- 400.8/567.8 MB 18.2 MB/s eta 0:00:10\n",
      "     -------------------------- ---------- 401.7/567.8 MB 18.2 MB/s eta 0:00:10\n",
      "     -------------------------- ---------- 402.4/567.8 MB 17.7 MB/s eta 0:00:10\n",
      "     -------------------------- ---------- 403.3/567.8 MB 17.2 MB/s eta 0:00:10\n",
      "     -------------------------- ---------- 404.1/567.8 MB 18.2 MB/s eta 0:00:09\n",
      "     -------------------------- ---------- 405.0/567.8 MB 17.7 MB/s eta 0:00:10\n",
      "     -------------------------- ---------- 405.9/567.8 MB 17.7 MB/s eta 0:00:10\n",
      "     -------------------------- ---------- 406.7/567.8 MB 17.7 MB/s eta 0:00:10\n",
      "     -------------------------- ---------- 407.6/567.8 MB 18.2 MB/s eta 0:00:09\n",
      "     -------------------------- ---------- 408.5/567.8 MB 18.2 MB/s eta 0:00:09\n",
      "     -------------------------- ---------- 409.3/567.8 MB 18.2 MB/s eta 0:00:09\n",
      "     -------------------------- ---------- 410.1/567.8 MB 18.2 MB/s eta 0:00:09\n",
      "     -------------------------- ---------- 411.0/567.8 MB 18.2 MB/s eta 0:00:09\n",
      "     -------------------------- ---------- 412.0/567.8 MB 18.7 MB/s eta 0:00:09\n",
      "     -------------------------- ---------- 412.9/567.8 MB 19.3 MB/s eta 0:00:09\n",
      "     -------------------------- ---------- 413.8/567.8 MB 19.3 MB/s eta 0:00:08\n",
      "     --------------------------- --------- 414.8/567.8 MB 19.2 MB/s eta 0:00:08\n",
      "     --------------------------- --------- 415.6/567.8 MB 19.2 MB/s eta 0:00:08\n",
      "     --------------------------- --------- 416.4/567.8 MB 18.7 MB/s eta 0:00:09\n",
      "     --------------------------- --------- 417.3/567.8 MB 18.7 MB/s eta 0:00:09\n",
      "     --------------------------- --------- 418.2/567.8 MB 19.3 MB/s eta 0:00:08\n",
      "     --------------------------- --------- 419.1/567.8 MB 19.3 MB/s eta 0:00:08\n",
      "     --------------------------- --------- 420.1/567.8 MB 19.3 MB/s eta 0:00:08\n",
      "     --------------------------- --------- 421.1/567.8 MB 19.8 MB/s eta 0:00:08\n",
      "     --------------------------- --------- 421.9/567.8 MB 19.8 MB/s eta 0:00:08\n",
      "     --------------------------- --------- 422.9/567.8 MB 19.8 MB/s eta 0:00:08\n",
      "     --------------------------- --------- 423.8/567.8 MB 19.3 MB/s eta 0:00:08\n",
      "     --------------------------- --------- 424.7/567.8 MB 19.9 MB/s eta 0:00:08\n",
      "     --------------------------- --------- 425.6/567.8 MB 19.9 MB/s eta 0:00:08\n",
      "     --------------------------- --------- 426.5/567.8 MB 19.8 MB/s eta 0:00:08\n",
      "     --------------------------- --------- 427.4/567.8 MB 19.8 MB/s eta 0:00:08\n",
      "     --------------------------- --------- 428.4/567.8 MB 19.8 MB/s eta 0:00:08\n",
      "     --------------------------- --------- 429.4/567.8 MB 19.8 MB/s eta 0:00:07\n",
      "     ---------------------------- -------- 430.3/567.8 MB 19.8 MB/s eta 0:00:07\n",
      "     ---------------------------- -------- 431.2/567.8 MB 19.9 MB/s eta 0:00:07\n",
      "     ---------------------------- -------- 432.1/567.8 MB 19.9 MB/s eta 0:00:07\n",
      "     ---------------------------- -------- 432.9/567.8 MB 19.3 MB/s eta 0:00:08\n",
      "     ---------------------------- -------- 433.8/567.8 MB 19.3 MB/s eta 0:00:07\n",
      "     ---------------------------- -------- 434.6/567.8 MB 19.3 MB/s eta 0:00:07\n",
      "     ---------------------------- -------- 435.4/567.8 MB 19.3 MB/s eta 0:00:07\n",
      "     ---------------------------- -------- 436.2/567.8 MB 18.7 MB/s eta 0:00:08\n",
      "     ---------------------------- -------- 437.3/567.8 MB 19.2 MB/s eta 0:00:07\n",
      "     ---------------------------- -------- 438.2/567.8 MB 19.3 MB/s eta 0:00:07\n",
      "     ---------------------------- -------- 439.3/567.8 MB 19.3 MB/s eta 0:00:07\n",
      "     ---------------------------- -------- 440.3/567.8 MB 19.3 MB/s eta 0:00:07\n",
      "     ---------------------------- -------- 441.3/567.8 MB 19.8 MB/s eta 0:00:07\n",
      "     ---------------------------- -------- 442.4/567.8 MB 19.8 MB/s eta 0:00:07\n",
      "     ---------------------------- -------- 443.3/567.8 MB 19.8 MB/s eta 0:00:07\n",
      "     ---------------------------- -------- 444.3/567.8 MB 20.5 MB/s eta 0:00:07\n",
      "     ----------------------------- ------- 445.4/567.8 MB 21.1 MB/s eta 0:00:06\n",
      "     ----------------------------- ------- 446.3/567.8 MB 21.9 MB/s eta 0:00:06\n",
      "     ----------------------------- ------- 447.3/567.8 MB 21.9 MB/s eta 0:00:06\n",
      "     ----------------------------- ------- 448.3/567.8 MB 21.1 MB/s eta 0:00:06\n",
      "     ----------------------------- ------- 449.3/567.8 MB 21.1 MB/s eta 0:00:06\n",
      "     ----------------------------- ------- 450.2/567.8 MB 21.1 MB/s eta 0:00:06\n",
      "     ----------------------------- ------- 451.1/567.8 MB 21.8 MB/s eta 0:00:06\n",
      "     ----------------------------- ------- 452.1/567.8 MB 21.1 MB/s eta 0:00:06\n",
      "     ----------------------------- ------- 453.0/567.8 MB 20.5 MB/s eta 0:00:06\n",
      "     ----------------------------- ------- 454.2/567.8 MB 21.1 MB/s eta 0:00:06\n",
      "     ----------------------------- ------- 455.1/567.8 MB 21.1 MB/s eta 0:00:06\n",
      "     ----------------------------- ------- 456.1/567.8 MB 21.1 MB/s eta 0:00:06\n",
      "     ----------------------------- ------- 457.1/567.8 MB 21.1 MB/s eta 0:00:06\n",
      "     ----------------------------- ------- 458.0/567.8 MB 20.5 MB/s eta 0:00:06\n",
      "     ----------------------------- ------- 458.9/567.8 MB 21.1 MB/s eta 0:00:06\n",
      "     ----------------------------- ------- 459.9/567.8 MB 20.5 MB/s eta 0:00:06\n",
      "     ------------------------------ ------ 460.8/567.8 MB 20.5 MB/s eta 0:00:06\n",
      "     ------------------------------ ------ 461.4/567.8 MB 19.9 MB/s eta 0:00:06\n",
      "     ------------------------------ ------ 462.1/567.8 MB 19.9 MB/s eta 0:00:06\n",
      "     ------------------------------ ------ 462.8/567.8 MB 18.7 MB/s eta 0:00:06\n",
      "     ------------------------------ ------ 463.7/567.8 MB 18.7 MB/s eta 0:00:06\n",
      "     ------------------------------ ------ 464.4/567.8 MB 18.2 MB/s eta 0:00:06\n",
      "     ------------------------------ ------ 465.2/567.8 MB 17.7 MB/s eta 0:00:06\n",
      "     ------------------------------ ------ 465.7/567.8 MB 17.7 MB/s eta 0:00:06\n",
      "     ------------------------------ ------ 466.4/567.8 MB 17.2 MB/s eta 0:00:06\n",
      "     ------------------------------ ------ 467.1/567.8 MB 16.4 MB/s eta 0:00:07\n",
      "     ------------------------------ ------ 468.0/567.8 MB 16.4 MB/s eta 0:00:07\n",
      "     ------------------------------ ------ 468.7/567.8 MB 16.0 MB/s eta 0:00:07\n",
      "     ------------------------------ ------ 469.5/567.8 MB 16.4 MB/s eta 0:00:07\n",
      "     ------------------------------ ------ 470.2/567.8 MB 16.0 MB/s eta 0:00:07\n",
      "     ------------------------------ ------ 470.9/567.8 MB 16.0 MB/s eta 0:00:07\n",
      "     ------------------------------ ------ 471.7/567.8 MB 16.0 MB/s eta 0:00:07\n",
      "     ------------------------------ ------ 472.5/567.8 MB 16.0 MB/s eta 0:00:06\n",
      "     ------------------------------ ------ 473.2/567.8 MB 16.4 MB/s eta 0:00:06\n",
      "     ------------------------------ ------ 473.9/567.8 MB 15.6 MB/s eta 0:00:07\n",
      "     ------------------------------ ------ 474.8/567.8 MB 16.0 MB/s eta 0:00:06\n",
      "     ------------------------------ ------ 475.5/567.8 MB 16.4 MB/s eta 0:00:06\n",
      "     ------------------------------- ----- 476.4/567.8 MB 16.4 MB/s eta 0:00:06\n",
      "     ------------------------------- ----- 477.2/567.8 MB 16.8 MB/s eta 0:00:06\n",
      "     ------------------------------- ----- 478.1/567.8 MB 16.4 MB/s eta 0:00:06\n",
      "     ------------------------------- ----- 478.9/567.8 MB 17.2 MB/s eta 0:00:06\n",
      "     ------------------------------- ----- 479.8/567.8 MB 17.2 MB/s eta 0:00:06\n",
      "     ------------------------------- ----- 480.7/567.8 MB 17.2 MB/s eta 0:00:06\n",
      "     ------------------------------- ----- 481.4/567.8 MB 17.7 MB/s eta 0:00:05\n",
      "     ------------------------------- ----- 482.2/567.8 MB 17.7 MB/s eta 0:00:05\n",
      "     ------------------------------- ----- 483.1/567.8 MB 17.7 MB/s eta 0:00:05\n",
      "     ------------------------------- ----- 483.9/567.8 MB 17.7 MB/s eta 0:00:05\n",
      "     ------------------------------- ----- 484.7/567.8 MB 18.2 MB/s eta 0:00:05\n",
      "     ------------------------------- ----- 485.6/567.8 MB 18.2 MB/s eta 0:00:05\n",
      "     ------------------------------- ----- 486.5/567.8 MB 17.7 MB/s eta 0:00:05\n",
      "     ------------------------------- ----- 487.3/567.8 MB 18.2 MB/s eta 0:00:05\n",
      "     ------------------------------- ----- 488.2/567.8 MB 18.2 MB/s eta 0:00:05\n",
      "     ------------------------------- ----- 489.0/567.8 MB 18.2 MB/s eta 0:00:05\n",
      "     ------------------------------- ----- 489.9/567.8 MB 18.2 MB/s eta 0:00:05\n",
      "     ------------------------------- ----- 490.8/567.8 MB 17.7 MB/s eta 0:00:05\n",
      "     -------------------------------- ---- 491.7/567.8 MB 18.2 MB/s eta 0:00:05\n",
      "     -------------------------------- ---- 492.5/567.8 MB 18.2 MB/s eta 0:00:05\n",
      "     -------------------------------- ---- 493.3/567.8 MB 17.7 MB/s eta 0:00:05\n",
      "     -------------------------------- ---- 494.2/567.8 MB 18.7 MB/s eta 0:00:04\n",
      "     -------------------------------- ---- 495.1/567.8 MB 18.7 MB/s eta 0:00:04\n",
      "     -------------------------------- ---- 496.0/567.8 MB 18.7 MB/s eta 0:00:04\n",
      "     -------------------------------- ---- 496.9/567.8 MB 18.7 MB/s eta 0:00:04\n",
      "     -------------------------------- ---- 497.8/567.8 MB 18.7 MB/s eta 0:00:04\n",
      "     -------------------------------- ---- 498.7/567.8 MB 19.3 MB/s eta 0:00:04\n",
      "     -------------------------------- ---- 499.7/567.8 MB 19.3 MB/s eta 0:00:04\n",
      "     -------------------------------- ---- 500.6/567.8 MB 19.3 MB/s eta 0:00:04\n",
      "     -------------------------------- ---- 501.5/567.8 MB 19.2 MB/s eta 0:00:04\n",
      "     -------------------------------- ---- 502.4/567.8 MB 19.8 MB/s eta 0:00:04\n",
      "     -------------------------------- ---- 503.3/567.8 MB 19.9 MB/s eta 0:00:04\n",
      "     -------------------------------- ---- 504.3/567.8 MB 19.9 MB/s eta 0:00:04\n",
      "     -------------------------------- ---- 505.1/567.8 MB 19.9 MB/s eta 0:00:04\n",
      "     -------------------------------- ---- 506.1/567.8 MB 19.8 MB/s eta 0:00:04\n",
      "     --------------------------------- --- 507.0/567.8 MB 19.8 MB/s eta 0:00:04\n",
      "     --------------------------------- --- 507.9/567.8 MB 19.8 MB/s eta 0:00:04\n",
      "     --------------------------------- --- 508.8/567.8 MB 19.8 MB/s eta 0:00:03\n",
      "     --------------------------------- --- 509.7/567.8 MB 19.8 MB/s eta 0:00:03\n",
      "     --------------------------------- --- 510.7/567.8 MB 19.9 MB/s eta 0:00:03\n",
      "     --------------------------------- --- 511.6/567.8 MB 19.9 MB/s eta 0:00:03\n",
      "     --------------------------------- --- 512.5/567.8 MB 19.9 MB/s eta 0:00:03\n",
      "     --------------------------------- --- 513.5/567.8 MB 19.8 MB/s eta 0:00:03\n",
      "     --------------------------------- --- 514.5/567.8 MB 19.8 MB/s eta 0:00:03\n",
      "     --------------------------------- --- 515.5/567.8 MB 19.8 MB/s eta 0:00:03\n",
      "     --------------------------------- --- 516.3/567.8 MB 19.8 MB/s eta 0:00:03\n",
      "     --------------------------------- --- 517.3/567.8 MB 19.8 MB/s eta 0:00:03\n",
      "     --------------------------------- --- 518.2/567.8 MB 19.9 MB/s eta 0:00:03\n",
      "     --------------------------------- --- 519.2/567.8 MB 19.9 MB/s eta 0:00:03\n",
      "     --------------------------------- --- 520.1/567.8 MB 19.9 MB/s eta 0:00:03\n",
      "     --------------------------------- --- 521.0/567.8 MB 19.8 MB/s eta 0:00:03\n",
      "     ---------------------------------- -- 522.0/567.8 MB 19.8 MB/s eta 0:00:03\n",
      "     ---------------------------------- -- 523.0/567.8 MB 20.5 MB/s eta 0:00:03\n",
      "     ---------------------------------- -- 523.9/567.8 MB 20.5 MB/s eta 0:00:03\n",
      "     ---------------------------------- -- 524.9/567.8 MB 20.5 MB/s eta 0:00:03\n",
      "     ---------------------------------- -- 525.8/567.8 MB 20.5 MB/s eta 0:00:03\n",
      "     ---------------------------------- -- 526.8/567.8 MB 20.5 MB/s eta 0:00:03\n",
      "     ---------------------------------- -- 527.7/567.8 MB 20.5 MB/s eta 0:00:02\n",
      "     ---------------------------------- -- 528.7/567.8 MB 20.5 MB/s eta 0:00:02\n",
      "     ---------------------------------- -- 529.5/567.8 MB 19.8 MB/s eta 0:00:02\n",
      "     ---------------------------------- -- 530.6/567.8 MB 20.5 MB/s eta 0:00:02\n",
      "     ---------------------------------- -- 531.4/567.8 MB 20.5 MB/s eta 0:00:02\n",
      "     ---------------------------------- -- 532.4/567.8 MB 20.5 MB/s eta 0:00:02\n",
      "     ---------------------------------- -- 533.4/567.8 MB 19.9 MB/s eta 0:00:02\n",
      "     ---------------------------------- -- 534.4/567.8 MB 20.5 MB/s eta 0:00:02\n",
      "     ---------------------------------- -- 535.4/567.8 MB 20.5 MB/s eta 0:00:02\n",
      "     ---------------------------------- -- 536.3/567.8 MB 19.8 MB/s eta 0:00:02\n",
      "     ----------------------------------- - 537.2/567.8 MB 20.5 MB/s eta 0:00:02\n",
      "     ----------------------------------- - 538.2/567.8 MB 20.5 MB/s eta 0:00:02\n",
      "     ----------------------------------- - 539.2/567.8 MB 21.1 MB/s eta 0:00:02\n",
      "     ----------------------------------- - 540.2/567.8 MB 21.1 MB/s eta 0:00:02\n",
      "     ----------------------------------- - 541.2/567.8 MB 21.1 MB/s eta 0:00:02\n",
      "     ----------------------------------- - 542.2/567.8 MB 21.1 MB/s eta 0:00:02\n",
      "     ----------------------------------- - 543.2/567.8 MB 21.1 MB/s eta 0:00:02\n",
      "     ----------------------------------- - 544.2/567.8 MB 21.9 MB/s eta 0:00:02\n",
      "     ----------------------------------- - 545.2/567.8 MB 21.1 MB/s eta 0:00:02\n",
      "     ----------------------------------- - 546.3/567.8 MB 21.8 MB/s eta 0:00:01\n",
      "     ----------------------------------- - 547.3/567.8 MB 21.8 MB/s eta 0:00:01\n",
      "     ----------------------------------- - 548.3/567.8 MB 21.8 MB/s eta 0:00:01\n",
      "     ----------------------------------- - 549.4/567.8 MB 21.8 MB/s eta 0:00:01\n",
      "     ----------------------------------- - 550.5/567.8 MB 22.6 MB/s eta 0:00:01\n",
      "     ----------------------------------- - 551.5/567.8 MB 22.5 MB/s eta 0:00:01\n",
      "     ----------------------------------- - 552.0/567.8 MB 21.8 MB/s eta 0:00:01\n",
      "     ------------------------------------  552.8/567.8 MB 21.8 MB/s eta 0:00:01\n",
      "     ------------------------------------  553.4/567.8 MB 19.9 MB/s eta 0:00:01\n",
      "     ------------------------------------  554.1/567.8 MB 19.3 MB/s eta 0:00:01\n",
      "     ------------------------------------  554.9/567.8 MB 19.3 MB/s eta 0:00:01\n",
      "     ------------------------------------  555.7/567.8 MB 18.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  556.5/567.8 MB 18.2 MB/s eta 0:00:01\n",
      "     ------------------------------------  557.3/567.8 MB 18.2 MB/s eta 0:00:01\n",
      "     ------------------------------------  558.0/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  558.7/567.8 MB 17.3 MB/s eta 0:00:01\n",
      "     ------------------------------------  559.4/567.8 MB 16.4 MB/s eta 0:00:01\n",
      "     ------------------------------------  560.2/567.8 MB 16.4 MB/s eta 0:00:01\n",
      "     ------------------------------------  561.0/567.8 MB 16.0 MB/s eta 0:00:01\n",
      "     ------------------------------------  562.0/567.8 MB 16.0 MB/s eta 0:00:01\n",
      "     ------------------------------------  562.7/567.8 MB 16.4 MB/s eta 0:00:01\n",
      "     ------------------------------------  563.6/567.8 MB 16.8 MB/s eta 0:00:01\n",
      "     ------------------------------------  564.4/567.8 MB 17.3 MB/s eta 0:00:01\n",
      "     ------------------------------------  565.2/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  566.1/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  566.9/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.7/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     ------------------------------------  567.8/567.8 MB 17.7 MB/s eta 0:00:01\n",
      "     -------------------------------------- 567.8/567.8 MB 2.2 MB/s eta 0:00:00\n",
      "Requirement already satisfied: spacy<3.8.0,>=3.7.0 in c:\\users\\algro\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from de-core-news-lg==3.7.0) (3.7.2)\n",
      "Requirement already satisfied: spacy-legacy<3.1.0,>=3.0.11 in c:\\users\\algro\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from spacy<3.8.0,>=3.7.0->de-core-news-lg==3.7.0) (3.0.12)\n",
      "Requirement already satisfied: spacy-loggers<2.0.0,>=1.0.0 in c:\\users\\algro\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from spacy<3.8.0,>=3.7.0->de-core-news-lg==3.7.0) (1.0.5)\n",
      "Requirement already satisfied: murmurhash<1.1.0,>=0.28.0 in c:\\users\\algro\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from spacy<3.8.0,>=3.7.0->de-core-news-lg==3.7.0) (1.0.10)\n",
      "Requirement already satisfied: cymem<2.1.0,>=2.0.2 in c:\\users\\algro\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from spacy<3.8.0,>=3.7.0->de-core-news-lg==3.7.0) (2.0.8)\n",
      "Requirement already satisfied: preshed<3.1.0,>=3.0.2 in c:\\users\\algro\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from spacy<3.8.0,>=3.7.0->de-core-news-lg==3.7.0) (3.0.9)\n",
      "Requirement already satisfied: thinc<8.3.0,>=8.1.8 in c:\\users\\algro\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from spacy<3.8.0,>=3.7.0->de-core-news-lg==3.7.0) (8.2.1)\n",
      "Requirement already satisfied: wasabi<1.2.0,>=0.9.1 in c:\\users\\algro\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from spacy<3.8.0,>=3.7.0->de-core-news-lg==3.7.0) (1.1.2)\n",
      "Requirement already satisfied: srsly<3.0.0,>=2.4.3 in c:\\users\\algro\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from spacy<3.8.0,>=3.7.0->de-core-news-lg==3.7.0) (2.4.8)\n",
      "Requirement already satisfied: catalogue<2.1.0,>=2.0.6 in c:\\users\\algro\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from spacy<3.8.0,>=3.7.0->de-core-news-lg==3.7.0) (2.0.10)\n",
      "Requirement already satisfied: weasel<0.4.0,>=0.1.0 in c:\\users\\algro\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from spacy<3.8.0,>=3.7.0->de-core-news-lg==3.7.0) (0.3.3)\n",
      "Requirement already satisfied: typer<0.10.0,>=0.3.0 in c:\\users\\algro\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from spacy<3.8.0,>=3.7.0->de-core-news-lg==3.7.0) (0.9.0)\n",
      "Requirement already satisfied: smart-open<7.0.0,>=5.2.1 in c:\\users\\algro\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from spacy<3.8.0,>=3.7.0->de-core-news-lg==3.7.0) (6.4.0)\n",
      "Requirement already satisfied: tqdm<5.0.0,>=4.38.0 in c:\\users\\algro\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from spacy<3.8.0,>=3.7.0->de-core-news-lg==3.7.0) (4.66.1)\n",
      "Requirement already satisfied: requests<3.0.0,>=2.13.0 in c:\\users\\algro\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from spacy<3.8.0,>=3.7.0->de-core-news-lg==3.7.0) (2.28.2)\n",
      "Requirement already satisfied: pydantic!=1.8,!=1.8.1,<3.0.0,>=1.7.4 in c:\\users\\algro\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from spacy<3.8.0,>=3.7.0->de-core-news-lg==3.7.0) (2.4.2)\n",
      "Requirement already satisfied: jinja2 in c:\\users\\algro\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from spacy<3.8.0,>=3.7.0->de-core-news-lg==3.7.0) (3.1.2)\n",
      "Requirement already satisfied: setuptools in c:\\users\\algro\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from spacy<3.8.0,>=3.7.0->de-core-news-lg==3.7.0) (65.5.0)\n",
      "Requirement already satisfied: packaging>=20.0 in c:\\users\\algro\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from spacy<3.8.0,>=3.7.0->de-core-news-lg==3.7.0) (23.0)\n",
      "Requirement already satisfied: langcodes<4.0.0,>=3.2.0 in c:\\users\\algro\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from spacy<3.8.0,>=3.7.0->de-core-news-lg==3.7.0) (3.3.0)\n",
      "Requirement already satisfied: numpy>=1.19.0 in c:\\users\\algro\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from spacy<3.8.0,>=3.7.0->de-core-news-lg==3.7.0) (1.24.2)\n",
      "Requirement already satisfied: annotated-types>=0.4.0 in c:\\users\\algro\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from pydantic!=1.8,!=1.8.1,<3.0.0,>=1.7.4->spacy<3.8.0,>=3.7.0->de-core-news-lg==3.7.0) (0.6.0)\n",
      "Requirement already satisfied: pydantic-core==2.10.1 in c:\\users\\algro\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from pydantic!=1.8,!=1.8.1,<3.0.0,>=1.7.4->spacy<3.8.0,>=3.7.0->de-core-news-lg==3.7.0) (2.10.1)\n",
      "Requirement already satisfied: typing-extensions>=4.6.1 in c:\\users\\algro\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from pydantic!=1.8,!=1.8.1,<3.0.0,>=1.7.4->spacy<3.8.0,>=3.7.0->de-core-news-lg==3.7.0) (4.8.0)\n",
      "Requirement already satisfied: charset-normalizer<4,>=2 in c:\\users\\algro\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from requests<3.0.0,>=2.13.0->spacy<3.8.0,>=3.7.0->de-core-news-lg==3.7.0) (3.0.1)\n",
      "Requirement already satisfied: idna<4,>=2.5 in c:\\users\\algro\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from requests<3.0.0,>=2.13.0->spacy<3.8.0,>=3.7.0->de-core-news-lg==3.7.0) (2.10)\n",
      "Requirement already satisfied: urllib3<1.27,>=1.21.1 in c:\\users\\algro\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from requests<3.0.0,>=2.13.0->spacy<3.8.0,>=3.7.0->de-core-news-lg==3.7.0) (1.26.14)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in c:\\users\\algro\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from requests<3.0.0,>=2.13.0->spacy<3.8.0,>=3.7.0->de-core-news-lg==3.7.0) (2022.12.7)\n",
      "Requirement already satisfied: blis<0.8.0,>=0.7.8 in c:\\users\\algro\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from thinc<8.3.0,>=8.1.8->spacy<3.8.0,>=3.7.0->de-core-news-lg==3.7.0) (0.7.11)\n",
      "Requirement already satisfied: confection<1.0.0,>=0.0.1 in c:\\users\\algro\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from thinc<8.3.0,>=8.1.8->spacy<3.8.0,>=3.7.0->de-core-news-lg==3.7.0) (0.1.3)\n",
      "Requirement already satisfied: colorama in c:\\users\\algro\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from tqdm<5.0.0,>=4.38.0->spacy<3.8.0,>=3.7.0->de-core-news-lg==3.7.0) (0.4.6)\n",
      "Requirement already satisfied: click<9.0.0,>=7.1.1 in c:\\users\\algro\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from typer<0.10.0,>=0.3.0->spacy<3.8.0,>=3.7.0->de-core-news-lg==3.7.0) (8.1.3)\n",
      "Requirement already satisfied: cloudpathlib<0.17.0,>=0.7.0 in c:\\users\\algro\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from weasel<0.4.0,>=0.1.0->spacy<3.8.0,>=3.7.0->de-core-news-lg==3.7.0) (0.16.0)\n",
      "Requirement already satisfied: MarkupSafe>=2.0 in c:\\users\\algro\\appdata\\local\\programs\\python\\python311\\lib\\site-packages (from jinja2->spacy<3.8.0,>=3.7.0->de-core-news-lg==3.7.0) (2.1.2)\n",
      "\u001b[38;5;2m✔ Download and installation successful\u001b[0m\n",
      "You can now load the package via spacy.load('de_core_news_lg')\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "[notice] A new release of pip is available: 23.3 -> 24.0\n",
      "[notice] To update, run: python.exe -m pip install --upgrade pip\n"
     ]
    }
   ],
   "source": [
    "# Downloading a German SpaCy Language Model\n",
    "!python -m spacy download de_core_news_lg"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Loading SpaCy Model and Defining Entity Extraction Function\n",
    "import spacy\n",
    "\n",
    "model = spacy.load(\"de_core_news_lg\")\n",
    "\n",
    "def get_entities(sentence):\n",
    "  doc = model(sentence)\n",
    "  ent_lst = []\n",
    "  for ent in doc.ents:\n",
    "    if ent.label_ != 'MISC':\n",
    "      ent_lst.append(ent) #  ent_lst.append((ent, ent.label_)) here the label is extracted with the word.. \n",
    "  return ent_lst"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Processing XML Files\n",
    "input_folder = 'DATA/tei'\n",
    "\n",
    "data_dict = {}\n",
    "\n",
    "# Loop through each file in the input folder and add data to a dictionary with the ID as a key\n",
    "for filename in os.listdir(input_folder):\n",
    "    if filename.endswith('.xml'):  # Process only XML files\n",
    "        input_file_path = os.path.join(input_folder, filename)\n",
    "        \n",
    "        temp_list = []\n",
    "\n",
    "        # Extract the metadata\n",
    "        tree = ET.parse(input_file_path)\n",
    "        root = tree.getroot()\n",
    "\n",
    "        date = root.find('.//date').text if root.find('.//date') is not None else \"\"\n",
    "        temp_list.append(date)\n",
    "\n",
    "        place = root.find('.//pubPlace').text if root.find('.//pubPlace') is not None else \"\"\n",
    "        temp_list.append(place)\n",
    "\n",
    "        # Extract the content\n",
    "        with open(input_file_path, 'r', encoding='utf-8') as f:\n",
    "            content = f.read()\n",
    "\n",
    "        # Replace \"<\" with \" <\" to add a space before XML tags\n",
    "        content = content.replace(\"<\", \" <\")\n",
    "        \n",
    "        # Replace \"=<\" with \"<\" to remove the space before XML tags\n",
    "        content = content.replace(\"= <\", \"<\")\n",
    "\n",
    "        soup = BeautifulSoup(content, 'xml')\n",
    "        text = soup.get_text()\n",
    "        \n",
    "        # Replace \"=\" with \"-\"\n",
    "        text = text.replace(\"=\", \"-\")\n",
    "\n",
    "        # Use regular expression to find and remove text before the first occurrence of \"Mannheim\" and remove leading/trailing whitespace\n",
    "        new_text = re.sub(r\"^(.*?Mannheim)\", \"\", text, flags=re.DOTALL)\n",
    "\n",
    "        # Remove some of the whitespace\n",
    "        new_string = new_text.strip().replace('\\n\\n', '\\n')\n",
    "        temp_list.append(new_string)\n",
    "\n",
    "        # Extract the named entities\n",
    "        ner_results = get_entities(new_string)\n",
    "        unique_tags = list(set(ner_results))\n",
    "        temp_list.append(unique_tags)\n",
    "\n",
    "\n",
    "        data_dict[filename[:8]] = temp_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         ID        Date              Place  \\\n",
      "0  khz01001  10-01-1744  Erfurt, Thüringen   \n",
      "1  khz01002  13-01-1744  Erfurt, Thüringen   \n",
      "2  khz01003  17-01-1744  Erfurt, Thüringen   \n",
      "3  khz01004  07-02-1744  Erfurt, Thüringen   \n",
      "4  khz01005  21-02-1744  Erfurt, Thüringen   \n",
      "\n",
      "                                             Content  \\\n",
      "0  Der allerneuesten \\n\\nEuropäischen \\n\\nWelt- u...   \n",
      "1  Der allerneuesten   Europäischen   Welt- und S...   \n",
      "2  Der allerneuesten   Europäischen   Welt- und S...   \n",
      "3  Der allerneuesten   Europäischen   Welt- und S...   \n",
      "4  Der allerneuesten   Europäischen   Welt-und St...   \n",
      "\n",
      "                                               Words  \n",
      "0  [(Clevischen), (Graf, von), (Holland), (hertzl...  \n",
      "1  [(Fürst, von, Lichtenstein), (Ubergangs), (Eng...  \n",
      "2  [(Amts,  ), (Holland), (Jtalien), (August, von...  \n",
      "3  [(Po), (Jhren, Sold), (Chur), (Pescaro), (D), ...  \n",
      "4  [(Frantzösischen, Escadern), (Canonicum), (hof...  \n"
     ]
    }
   ],
   "source": [
    "# Creating DataFrame and Modifying Index\n",
    "df = pd.DataFrame.from_dict(data_dict, orient='index', columns=['Date', 'Place', 'Content', 'Words']).reset_index()\n",
    "\n",
    "df = df.rename(columns={'index': 'ID'})\n",
    "\n",
    "print(df.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Removing All Entries Without Proper Date Format\n",
    "df_new = df[df['Date'].astype(str).str.len() == 10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "object\n",
      "datetime64[ns]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\algro\\AppData\\Local\\Temp\\ipykernel_25312\\2986084493.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_new['Date'] = pd.to_datetime(df_new['Date'], format='%d-%m-%Y')\n"
     ]
    }
   ],
   "source": [
    "# Converting Date Format and Extracting Year and Day-Month\n",
    "print(df_new['Date'].dtype)\n",
    "df_new['Date'] = pd.to_datetime(df_new['Date'], format='%d-%m-%Y')\n",
    "print(df_new['Date'].dtype)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\algro\\AppData\\Local\\Temp\\ipykernel_25312\\801268230.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_new['Year'] = df_new['Date'].dt.year\n",
      "C:\\Users\\algro\\AppData\\Local\\Temp\\ipykernel_25312\\801268230.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_new['Day_month'] = df_new['Date'].dt.strftime('%d-%m')\n"
     ]
    }
   ],
   "source": [
    "df_new['Year'] = df_new['Date'].dt.year\n",
    "df_new['Day_month'] = df_new['Date'].dt.strftime('%d-%m')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\algro\\AppData\\Local\\Temp\\ipykernel_25312\\2974257692.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_new['Day'] = split_values[0]\n",
      "C:\\Users\\algro\\AppData\\Local\\Temp\\ipykernel_25312\\2974257692.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_new['Month'] = split_values[1]\n"
     ]
    }
   ],
   "source": [
    "# Splitting Day and Month\n",
    "split_values = df_new['Day_month'].str.split('-', expand=True)\n",
    "df_new['Day'] = split_values[0]\n",
    "df_new['Month'] = split_values[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\algro\\AppData\\Local\\Temp\\ipykernel_25312\\1550253169.py:9: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_new['Place_corrected'] = df_new['Place'].apply(geo)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         ID       Date              Place  \\\n",
      "0  khz01001 1744-01-10  Erfurt, Thüringen   \n",
      "1  khz01002 1744-01-13  Erfurt, Thüringen   \n",
      "2  khz01003 1744-01-17  Erfurt, Thüringen   \n",
      "3  khz01004 1744-02-07  Erfurt, Thüringen   \n",
      "4  khz01005 1744-02-21  Erfurt, Thüringen   \n",
      "\n",
      "                                             Content  \\\n",
      "0  Der allerneuesten \\n\\nEuropäischen \\n\\nWelt- u...   \n",
      "1  Der allerneuesten   Europäischen   Welt- und S...   \n",
      "2  Der allerneuesten   Europäischen   Welt- und S...   \n",
      "3  Der allerneuesten   Europäischen   Welt- und S...   \n",
      "4  Der allerneuesten   Europäischen   Welt-und St...   \n",
      "\n",
      "                                               Words  Year Day_month Day  \\\n",
      "0  [(Clevischen), (Graf, von), (Holland), (hertzl...  1744     10-01  10   \n",
      "1  [(Fürst, von, Lichtenstein), (Ubergangs), (Eng...  1744     13-01  13   \n",
      "2  [(Amts,  ), (Holland), (Jtalien), (August, von...  1744     17-01  17   \n",
      "3  [(Po), (Jhren, Sold), (Chur), (Pescaro), (D), ...  1744     07-02  07   \n",
      "4  [(Frantzösischen, Escadern), (Canonicum), (hof...  1744     21-02  21   \n",
      "\n",
      "  Month    Place_corrected  \n",
      "0    01  Erfurt, Thüringen  \n",
      "1    01  Erfurt, Thüringen  \n",
      "2    01  Erfurt, Thüringen  \n",
      "3    02  Erfurt, Thüringen  \n",
      "4    02  Erfurt, Thüringen  \n"
     ]
    }
   ],
   "source": [
    "# Geographic Data Correction\n",
    "def geo (place):\n",
    "    if place == 'Leibzig, Sachsen':\n",
    "        return 'Leipzig, Sachsen'\n",
    "    elif place == 'Stuttgart und Tübingen, Württemberg':\n",
    "        return 'Stuttgart, Württemberg'\n",
    "    else: \n",
    "        return place\n",
    "\n",
    "df_new['Place_corrected'] = df_new['Place'].apply(geo)\n",
    "\n",
    "print(df_new.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Erfurt, Thüringen' 'Rudolstadt, Thüringen' 'Augsburg, Bayern'\n",
      " 'Frankfurt, Hessen' 'Würzburg, Bayern' 'Bayreuth, Bayern'\n",
      " 'Berlin, Preussen' 'Breslau, Schlesien' 'Leipzig, Sachsen' 'Bremen'\n",
      " 'Mainz' 'Stuttgart, Württemberg' 'Hattingen, Nordrhein-Westfalen'\n",
      " 'Berlin' 'Tübingen, Württemberg' 'Wien, Österreich']\n"
     ]
    }
   ],
   "source": [
    "places = df_new['Place_corrected'].unique()\n",
    "print(places)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Using Geopy for Geographic Coordinates\n",
    "from geopy.geocoders import Nominatim\n",
    "from geopy.extra.rate_limiter import RateLimiter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "RateLimiter caught an error, retrying (0/2 tries). Called with (*('Erfurt, Thüringen',), **{}).\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connection.py\", line 174, in _new_conn\n",
      "    conn = connection.create_connection(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\util\\connection.py\", line 95, in create_connection\n",
      "    raise err\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\util\\connection.py\", line 85, in create_connection\n",
      "    sock.connect(sa)\n",
      "TimeoutError: timed out\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 703, in urlopen\n",
      "    httplib_response = self._make_request(\n",
      "                       ^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 386, in _make_request\n",
      "    self._validate_conn(conn)\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 1042, in _validate_conn\n",
      "    conn.connect()\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connection.py\", line 358, in connect\n",
      "    self.sock = conn = self._new_conn()\n",
      "                       ^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connection.py\", line 179, in _new_conn\n",
      "    raise ConnectTimeoutError(\n",
      "urllib3.exceptions.ConnectTimeoutError: (<urllib3.connection.HTTPSConnection object at 0x0000018FFC82DA90>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)')\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\adapters.py\", line 489, in send\n",
      "    resp = conn.urlopen(\n",
      "           ^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 815, in urlopen\n",
      "    return self.urlopen(\n",
      "           ^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 815, in urlopen\n",
      "    return self.urlopen(\n",
      "           ^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 787, in urlopen\n",
      "    retries = retries.increment(\n",
      "              ^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\util\\retry.py\", line 592, in increment\n",
      "    raise MaxRetryError(_pool, url, error or ResponseError(cause))\n",
      "urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /search?q=Erfurt%2C+Th%C3%BCringen&format=json&limit=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000018FFC82DA90>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\adapters.py\", line 482, in _request\n",
      "    resp = self.session.get(url, timeout=timeout, headers=headers)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\sessions.py\", line 600, in get\n",
      "    return self.request(\"GET\", url, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\sessions.py\", line 587, in request\n",
      "    resp = self.send(prep, **send_kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\sessions.py\", line 701, in send\n",
      "    r = adapter.send(request, **kwargs)\n",
      "        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\adapters.py\", line 553, in send\n",
      "    raise ConnectTimeout(e, request=request)\n",
      "requests.exceptions.ConnectTimeout: HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /search?q=Erfurt%2C+Th%C3%BCringen&format=json&limit=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000018FFC82DA90>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\extra\\rate_limiter.py\", line 136, in _retries_gen\n",
      "    yield i  # Run the function.\n",
      "    ^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\extra\\rate_limiter.py\", line 274, in __call__\n",
      "    res = self.func(*args, **kwargs)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\geocoders\\nominatim.py\", line 297, in geocode\n",
      "    return self._call_geocoder(url, callback, timeout=timeout)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\geocoders\\base.py\", line 368, in _call_geocoder\n",
      "    result = self.adapter.get_json(url, timeout=timeout, headers=req_headers)\n",
      "             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\adapters.py\", line 472, in get_json\n",
      "    resp = self._request(url, timeout=timeout, headers=headers)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\adapters.py\", line 494, in _request\n",
      "    raise GeocoderUnavailable(message)\n",
      "geopy.exc.GeocoderUnavailable: HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /search?q=Erfurt%2C+Th%C3%BCringen&format=json&limit=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000018FFC82DA90>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "RateLimiter caught an error, retrying (1/2 tries). Called with (*('Erfurt, Thüringen',), **{}).\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connection.py\", line 174, in _new_conn\n",
      "    conn = connection.create_connection(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\util\\connection.py\", line 95, in create_connection\n",
      "    raise err\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\util\\connection.py\", line 85, in create_connection\n",
      "    sock.connect(sa)\n",
      "TimeoutError: timed out\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 703, in urlopen\n",
      "    httplib_response = self._make_request(\n",
      "                       ^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 386, in _make_request\n",
      "    self._validate_conn(conn)\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 1042, in _validate_conn\n",
      "    conn.connect()\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connection.py\", line 358, in connect\n",
      "    self.sock = conn = self._new_conn()\n",
      "                       ^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connection.py\", line 179, in _new_conn\n",
      "    raise ConnectTimeoutError(\n",
      "urllib3.exceptions.ConnectTimeoutError: (<urllib3.connection.HTTPSConnection object at 0x0000018FBF55A210>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)')\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\adapters.py\", line 489, in send\n",
      "    resp = conn.urlopen(\n",
      "           ^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 815, in urlopen\n",
      "    return self.urlopen(\n",
      "           ^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 815, in urlopen\n",
      "    return self.urlopen(\n",
      "           ^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 787, in urlopen\n",
      "    retries = retries.increment(\n",
      "              ^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\util\\retry.py\", line 592, in increment\n",
      "    raise MaxRetryError(_pool, url, error or ResponseError(cause))\n",
      "urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /search?q=Erfurt%2C+Th%C3%BCringen&format=json&limit=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000018FBF55A210>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\adapters.py\", line 482, in _request\n",
      "    resp = self.session.get(url, timeout=timeout, headers=headers)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\sessions.py\", line 600, in get\n",
      "    return self.request(\"GET\", url, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\sessions.py\", line 587, in request\n",
      "    resp = self.send(prep, **send_kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\sessions.py\", line 701, in send\n",
      "    r = adapter.send(request, **kwargs)\n",
      "        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\adapters.py\", line 553, in send\n",
      "    raise ConnectTimeout(e, request=request)\n",
      "requests.exceptions.ConnectTimeout: HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /search?q=Erfurt%2C+Th%C3%BCringen&format=json&limit=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000018FBF55A210>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\extra\\rate_limiter.py\", line 136, in _retries_gen\n",
      "    yield i  # Run the function.\n",
      "    ^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\extra\\rate_limiter.py\", line 274, in __call__\n",
      "    res = self.func(*args, **kwargs)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\geocoders\\nominatim.py\", line 297, in geocode\n",
      "    return self._call_geocoder(url, callback, timeout=timeout)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\geocoders\\base.py\", line 368, in _call_geocoder\n",
      "    result = self.adapter.get_json(url, timeout=timeout, headers=req_headers)\n",
      "             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\adapters.py\", line 472, in get_json\n",
      "    resp = self._request(url, timeout=timeout, headers=headers)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\adapters.py\", line 494, in _request\n",
      "    raise GeocoderUnavailable(message)\n",
      "geopy.exc.GeocoderUnavailable: HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /search?q=Erfurt%2C+Th%C3%BCringen&format=json&limit=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000018FBF55A210>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "RateLimiter swallowed an error after 2 retries. Called with (*('Erfurt, Thüringen',), **{}).\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connection.py\", line 174, in _new_conn\n",
      "    conn = connection.create_connection(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\util\\connection.py\", line 95, in create_connection\n",
      "    raise err\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\util\\connection.py\", line 85, in create_connection\n",
      "    sock.connect(sa)\n",
      "TimeoutError: timed out\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 703, in urlopen\n",
      "    httplib_response = self._make_request(\n",
      "                       ^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 386, in _make_request\n",
      "    self._validate_conn(conn)\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 1042, in _validate_conn\n",
      "    conn.connect()\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connection.py\", line 358, in connect\n",
      "    self.sock = conn = self._new_conn()\n",
      "                       ^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connection.py\", line 179, in _new_conn\n",
      "    raise ConnectTimeoutError(\n",
      "urllib3.exceptions.ConnectTimeoutError: (<urllib3.connection.HTTPSConnection object at 0x0000018F8A7EEF10>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)')\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\adapters.py\", line 489, in send\n",
      "    resp = conn.urlopen(\n",
      "           ^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 815, in urlopen\n",
      "    return self.urlopen(\n",
      "           ^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 815, in urlopen\n",
      "    return self.urlopen(\n",
      "           ^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 787, in urlopen\n",
      "    retries = retries.increment(\n",
      "              ^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\util\\retry.py\", line 592, in increment\n",
      "    raise MaxRetryError(_pool, url, error or ResponseError(cause))\n",
      "urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /search?q=Erfurt%2C+Th%C3%BCringen&format=json&limit=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000018F8A7EEF10>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\adapters.py\", line 482, in _request\n",
      "    resp = self.session.get(url, timeout=timeout, headers=headers)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\sessions.py\", line 600, in get\n",
      "    return self.request(\"GET\", url, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\sessions.py\", line 587, in request\n",
      "    resp = self.send(prep, **send_kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\sessions.py\", line 701, in send\n",
      "    r = adapter.send(request, **kwargs)\n",
      "        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\adapters.py\", line 553, in send\n",
      "    raise ConnectTimeout(e, request=request)\n",
      "requests.exceptions.ConnectTimeout: HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /search?q=Erfurt%2C+Th%C3%BCringen&format=json&limit=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000018F8A7EEF10>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\extra\\rate_limiter.py\", line 136, in _retries_gen\n",
      "    yield i  # Run the function.\n",
      "    ^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\extra\\rate_limiter.py\", line 274, in __call__\n",
      "    res = self.func(*args, **kwargs)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\geocoders\\nominatim.py\", line 297, in geocode\n",
      "    return self._call_geocoder(url, callback, timeout=timeout)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\geocoders\\base.py\", line 368, in _call_geocoder\n",
      "    result = self.adapter.get_json(url, timeout=timeout, headers=req_headers)\n",
      "             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\adapters.py\", line 472, in get_json\n",
      "    resp = self._request(url, timeout=timeout, headers=headers)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\adapters.py\", line 494, in _request\n",
      "    raise GeocoderUnavailable(message)\n",
      "geopy.exc.GeocoderUnavailable: HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /search?q=Erfurt%2C+Th%C3%BCringen&format=json&limit=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000018F8A7EEF10>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "RateLimiter caught an error, retrying (0/2 tries). Called with (*('Rudolstadt, Thüringen',), **{}).\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connection.py\", line 174, in _new_conn\n",
      "    conn = connection.create_connection(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\util\\connection.py\", line 95, in create_connection\n",
      "    raise err\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\util\\connection.py\", line 85, in create_connection\n",
      "    sock.connect(sa)\n",
      "TimeoutError: timed out\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 703, in urlopen\n",
      "    httplib_response = self._make_request(\n",
      "                       ^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 386, in _make_request\n",
      "    self._validate_conn(conn)\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 1042, in _validate_conn\n",
      "    conn.connect()\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connection.py\", line 358, in connect\n",
      "    self.sock = conn = self._new_conn()\n",
      "                       ^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connection.py\", line 179, in _new_conn\n",
      "    raise ConnectTimeoutError(\n",
      "urllib3.exceptions.ConnectTimeoutError: (<urllib3.connection.HTTPSConnection object at 0x0000018F8A9FA9D0>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)')\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\adapters.py\", line 489, in send\n",
      "    resp = conn.urlopen(\n",
      "           ^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 815, in urlopen\n",
      "    return self.urlopen(\n",
      "           ^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 815, in urlopen\n",
      "    return self.urlopen(\n",
      "           ^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 787, in urlopen\n",
      "    retries = retries.increment(\n",
      "              ^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\util\\retry.py\", line 592, in increment\n",
      "    raise MaxRetryError(_pool, url, error or ResponseError(cause))\n",
      "urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /search?q=Rudolstadt%2C+Th%C3%BCringen&format=json&limit=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000018F8A9FA9D0>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\adapters.py\", line 482, in _request\n",
      "    resp = self.session.get(url, timeout=timeout, headers=headers)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\sessions.py\", line 600, in get\n",
      "    return self.request(\"GET\", url, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\sessions.py\", line 587, in request\n",
      "    resp = self.send(prep, **send_kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\sessions.py\", line 701, in send\n",
      "    r = adapter.send(request, **kwargs)\n",
      "        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\adapters.py\", line 553, in send\n",
      "    raise ConnectTimeout(e, request=request)\n",
      "requests.exceptions.ConnectTimeout: HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /search?q=Rudolstadt%2C+Th%C3%BCringen&format=json&limit=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000018F8A9FA9D0>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\extra\\rate_limiter.py\", line 136, in _retries_gen\n",
      "    yield i  # Run the function.\n",
      "    ^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\extra\\rate_limiter.py\", line 274, in __call__\n",
      "    res = self.func(*args, **kwargs)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\geocoders\\nominatim.py\", line 297, in geocode\n",
      "    return self._call_geocoder(url, callback, timeout=timeout)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\geocoders\\base.py\", line 368, in _call_geocoder\n",
      "    result = self.adapter.get_json(url, timeout=timeout, headers=req_headers)\n",
      "             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\adapters.py\", line 472, in get_json\n",
      "    resp = self._request(url, timeout=timeout, headers=headers)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\adapters.py\", line 494, in _request\n",
      "    raise GeocoderUnavailable(message)\n",
      "geopy.exc.GeocoderUnavailable: HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /search?q=Rudolstadt%2C+Th%C3%BCringen&format=json&limit=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000018F8A9FA9D0>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "RateLimiter caught an error, retrying (1/2 tries). Called with (*('Rudolstadt, Thüringen',), **{}).\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connection.py\", line 174, in _new_conn\n",
      "    conn = connection.create_connection(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\util\\connection.py\", line 95, in create_connection\n",
      "    raise err\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\util\\connection.py\", line 85, in create_connection\n",
      "    sock.connect(sa)\n",
      "TimeoutError: timed out\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 703, in urlopen\n",
      "    httplib_response = self._make_request(\n",
      "                       ^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 386, in _make_request\n",
      "    self._validate_conn(conn)\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 1042, in _validate_conn\n",
      "    conn.connect()\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connection.py\", line 358, in connect\n",
      "    self.sock = conn = self._new_conn()\n",
      "                       ^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connection.py\", line 179, in _new_conn\n",
      "    raise ConnectTimeoutError(\n",
      "urllib3.exceptions.ConnectTimeoutError: (<urllib3.connection.HTTPSConnection object at 0x0000018F8A7EEA10>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)')\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\adapters.py\", line 489, in send\n",
      "    resp = conn.urlopen(\n",
      "           ^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 815, in urlopen\n",
      "    return self.urlopen(\n",
      "           ^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 815, in urlopen\n",
      "    return self.urlopen(\n",
      "           ^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 787, in urlopen\n",
      "    retries = retries.increment(\n",
      "              ^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\util\\retry.py\", line 592, in increment\n",
      "    raise MaxRetryError(_pool, url, error or ResponseError(cause))\n",
      "urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /search?q=Rudolstadt%2C+Th%C3%BCringen&format=json&limit=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000018F8A7EEA10>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\adapters.py\", line 482, in _request\n",
      "    resp = self.session.get(url, timeout=timeout, headers=headers)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\sessions.py\", line 600, in get\n",
      "    return self.request(\"GET\", url, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\sessions.py\", line 587, in request\n",
      "    resp = self.send(prep, **send_kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\sessions.py\", line 701, in send\n",
      "    r = adapter.send(request, **kwargs)\n",
      "        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\adapters.py\", line 553, in send\n",
      "    raise ConnectTimeout(e, request=request)\n",
      "requests.exceptions.ConnectTimeout: HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /search?q=Rudolstadt%2C+Th%C3%BCringen&format=json&limit=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000018F8A7EEA10>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\extra\\rate_limiter.py\", line 136, in _retries_gen\n",
      "    yield i  # Run the function.\n",
      "    ^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\extra\\rate_limiter.py\", line 274, in __call__\n",
      "    res = self.func(*args, **kwargs)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\geocoders\\nominatim.py\", line 297, in geocode\n",
      "    return self._call_geocoder(url, callback, timeout=timeout)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\geocoders\\base.py\", line 368, in _call_geocoder\n",
      "    result = self.adapter.get_json(url, timeout=timeout, headers=req_headers)\n",
      "             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\adapters.py\", line 472, in get_json\n",
      "    resp = self._request(url, timeout=timeout, headers=headers)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\adapters.py\", line 494, in _request\n",
      "    raise GeocoderUnavailable(message)\n",
      "geopy.exc.GeocoderUnavailable: HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /search?q=Rudolstadt%2C+Th%C3%BCringen&format=json&limit=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000018F8A7EEA10>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "RateLimiter swallowed an error after 2 retries. Called with (*('Rudolstadt, Thüringen',), **{}).\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connection.py\", line 174, in _new_conn\n",
      "    conn = connection.create_connection(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\util\\connection.py\", line 95, in create_connection\n",
      "    raise err\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\util\\connection.py\", line 85, in create_connection\n",
      "    sock.connect(sa)\n",
      "TimeoutError: timed out\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 703, in urlopen\n",
      "    httplib_response = self._make_request(\n",
      "                       ^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 386, in _make_request\n",
      "    self._validate_conn(conn)\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 1042, in _validate_conn\n",
      "    conn.connect()\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connection.py\", line 358, in connect\n",
      "    self.sock = conn = self._new_conn()\n",
      "                       ^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connection.py\", line 179, in _new_conn\n",
      "    raise ConnectTimeoutError(\n",
      "urllib3.exceptions.ConnectTimeoutError: (<urllib3.connection.HTTPSConnection object at 0x00000190541B1690>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)')\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\adapters.py\", line 489, in send\n",
      "    resp = conn.urlopen(\n",
      "           ^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 815, in urlopen\n",
      "    return self.urlopen(\n",
      "           ^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 815, in urlopen\n",
      "    return self.urlopen(\n",
      "           ^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 787, in urlopen\n",
      "    retries = retries.increment(\n",
      "              ^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\util\\retry.py\", line 592, in increment\n",
      "    raise MaxRetryError(_pool, url, error or ResponseError(cause))\n",
      "urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /search?q=Rudolstadt%2C+Th%C3%BCringen&format=json&limit=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x00000190541B1690>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\adapters.py\", line 482, in _request\n",
      "    resp = self.session.get(url, timeout=timeout, headers=headers)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\sessions.py\", line 600, in get\n",
      "    return self.request(\"GET\", url, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\sessions.py\", line 587, in request\n",
      "    resp = self.send(prep, **send_kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\sessions.py\", line 701, in send\n",
      "    r = adapter.send(request, **kwargs)\n",
      "        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\adapters.py\", line 553, in send\n",
      "    raise ConnectTimeout(e, request=request)\n",
      "requests.exceptions.ConnectTimeout: HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /search?q=Rudolstadt%2C+Th%C3%BCringen&format=json&limit=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x00000190541B1690>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\extra\\rate_limiter.py\", line 136, in _retries_gen\n",
      "    yield i  # Run the function.\n",
      "    ^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\extra\\rate_limiter.py\", line 274, in __call__\n",
      "    res = self.func(*args, **kwargs)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\geocoders\\nominatim.py\", line 297, in geocode\n",
      "    return self._call_geocoder(url, callback, timeout=timeout)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\geocoders\\base.py\", line 368, in _call_geocoder\n",
      "    result = self.adapter.get_json(url, timeout=timeout, headers=req_headers)\n",
      "             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\adapters.py\", line 472, in get_json\n",
      "    resp = self._request(url, timeout=timeout, headers=headers)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\adapters.py\", line 494, in _request\n",
      "    raise GeocoderUnavailable(message)\n",
      "geopy.exc.GeocoderUnavailable: HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /search?q=Rudolstadt%2C+Th%C3%BCringen&format=json&limit=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x00000190541B1690>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "RateLimiter caught an error, retrying (0/2 tries). Called with (*('Augsburg, Bayern',), **{}).\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connection.py\", line 174, in _new_conn\n",
      "    conn = connection.create_connection(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\util\\connection.py\", line 95, in create_connection\n",
      "    raise err\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\util\\connection.py\", line 85, in create_connection\n",
      "    sock.connect(sa)\n",
      "TimeoutError: timed out\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 703, in urlopen\n",
      "    httplib_response = self._make_request(\n",
      "                       ^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 386, in _make_request\n",
      "    self._validate_conn(conn)\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 1042, in _validate_conn\n",
      "    conn.connect()\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connection.py\", line 358, in connect\n",
      "    self.sock = conn = self._new_conn()\n",
      "                       ^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connection.py\", line 179, in _new_conn\n",
      "    raise ConnectTimeoutError(\n",
      "urllib3.exceptions.ConnectTimeoutError: (<urllib3.connection.HTTPSConnection object at 0x0000019039DA0510>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)')\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\adapters.py\", line 489, in send\n",
      "    resp = conn.urlopen(\n",
      "           ^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 815, in urlopen\n",
      "    return self.urlopen(\n",
      "           ^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 815, in urlopen\n",
      "    return self.urlopen(\n",
      "           ^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 787, in urlopen\n",
      "    retries = retries.increment(\n",
      "              ^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\util\\retry.py\", line 592, in increment\n",
      "    raise MaxRetryError(_pool, url, error or ResponseError(cause))\n",
      "urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /search?q=Augsburg%2C+Bayern&format=json&limit=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000019039DA0510>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\adapters.py\", line 482, in _request\n",
      "    resp = self.session.get(url, timeout=timeout, headers=headers)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\sessions.py\", line 600, in get\n",
      "    return self.request(\"GET\", url, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\sessions.py\", line 587, in request\n",
      "    resp = self.send(prep, **send_kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\sessions.py\", line 701, in send\n",
      "    r = adapter.send(request, **kwargs)\n",
      "        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\adapters.py\", line 553, in send\n",
      "    raise ConnectTimeout(e, request=request)\n",
      "requests.exceptions.ConnectTimeout: HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /search?q=Augsburg%2C+Bayern&format=json&limit=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000019039DA0510>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\extra\\rate_limiter.py\", line 136, in _retries_gen\n",
      "    yield i  # Run the function.\n",
      "    ^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\extra\\rate_limiter.py\", line 274, in __call__\n",
      "    res = self.func(*args, **kwargs)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\geocoders\\nominatim.py\", line 297, in geocode\n",
      "    return self._call_geocoder(url, callback, timeout=timeout)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\geocoders\\base.py\", line 368, in _call_geocoder\n",
      "    result = self.adapter.get_json(url, timeout=timeout, headers=req_headers)\n",
      "             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\adapters.py\", line 472, in get_json\n",
      "    resp = self._request(url, timeout=timeout, headers=headers)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\adapters.py\", line 494, in _request\n",
      "    raise GeocoderUnavailable(message)\n",
      "geopy.exc.GeocoderUnavailable: HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /search?q=Augsburg%2C+Bayern&format=json&limit=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000019039DA0510>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "RateLimiter caught an error, retrying (1/2 tries). Called with (*('Augsburg, Bayern',), **{}).\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connection.py\", line 174, in _new_conn\n",
      "    conn = connection.create_connection(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\util\\connection.py\", line 95, in create_connection\n",
      "    raise err\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\util\\connection.py\", line 85, in create_connection\n",
      "    sock.connect(sa)\n",
      "TimeoutError: timed out\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 703, in urlopen\n",
      "    httplib_response = self._make_request(\n",
      "                       ^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 386, in _make_request\n",
      "    self._validate_conn(conn)\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 1042, in _validate_conn\n",
      "    conn.connect()\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connection.py\", line 358, in connect\n",
      "    self.sock = conn = self._new_conn()\n",
      "                       ^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connection.py\", line 179, in _new_conn\n",
      "    raise ConnectTimeoutError(\n",
      "urllib3.exceptions.ConnectTimeoutError: (<urllib3.connection.HTTPSConnection object at 0x000001904E0D7050>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)')\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\adapters.py\", line 489, in send\n",
      "    resp = conn.urlopen(\n",
      "           ^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 815, in urlopen\n",
      "    return self.urlopen(\n",
      "           ^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 815, in urlopen\n",
      "    return self.urlopen(\n",
      "           ^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 787, in urlopen\n",
      "    retries = retries.increment(\n",
      "              ^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\util\\retry.py\", line 592, in increment\n",
      "    raise MaxRetryError(_pool, url, error or ResponseError(cause))\n",
      "urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /search?q=Augsburg%2C+Bayern&format=json&limit=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x000001904E0D7050>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\adapters.py\", line 482, in _request\n",
      "    resp = self.session.get(url, timeout=timeout, headers=headers)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\sessions.py\", line 600, in get\n",
      "    return self.request(\"GET\", url, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\sessions.py\", line 587, in request\n",
      "    resp = self.send(prep, **send_kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\sessions.py\", line 701, in send\n",
      "    r = adapter.send(request, **kwargs)\n",
      "        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\adapters.py\", line 553, in send\n",
      "    raise ConnectTimeout(e, request=request)\n",
      "requests.exceptions.ConnectTimeout: HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /search?q=Augsburg%2C+Bayern&format=json&limit=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x000001904E0D7050>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\extra\\rate_limiter.py\", line 136, in _retries_gen\n",
      "    yield i  # Run the function.\n",
      "    ^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\extra\\rate_limiter.py\", line 274, in __call__\n",
      "    res = self.func(*args, **kwargs)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\geocoders\\nominatim.py\", line 297, in geocode\n",
      "    return self._call_geocoder(url, callback, timeout=timeout)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\geocoders\\base.py\", line 368, in _call_geocoder\n",
      "    result = self.adapter.get_json(url, timeout=timeout, headers=req_headers)\n",
      "             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\adapters.py\", line 472, in get_json\n",
      "    resp = self._request(url, timeout=timeout, headers=headers)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\adapters.py\", line 494, in _request\n",
      "    raise GeocoderUnavailable(message)\n",
      "geopy.exc.GeocoderUnavailable: HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /search?q=Augsburg%2C+Bayern&format=json&limit=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x000001904E0D7050>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "RateLimiter swallowed an error after 2 retries. Called with (*('Augsburg, Bayern',), **{}).\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connection.py\", line 174, in _new_conn\n",
      "    conn = connection.create_connection(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\util\\connection.py\", line 95, in create_connection\n",
      "    raise err\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\util\\connection.py\", line 85, in create_connection\n",
      "    sock.connect(sa)\n",
      "TimeoutError: timed out\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 703, in urlopen\n",
      "    httplib_response = self._make_request(\n",
      "                       ^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 386, in _make_request\n",
      "    self._validate_conn(conn)\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 1042, in _validate_conn\n",
      "    conn.connect()\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connection.py\", line 358, in connect\n",
      "    self.sock = conn = self._new_conn()\n",
      "                       ^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connection.py\", line 179, in _new_conn\n",
      "    raise ConnectTimeoutError(\n",
      "urllib3.exceptions.ConnectTimeoutError: (<urllib3.connection.HTTPSConnection object at 0x0000018FC1B814D0>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)')\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\adapters.py\", line 489, in send\n",
      "    resp = conn.urlopen(\n",
      "           ^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 815, in urlopen\n",
      "    return self.urlopen(\n",
      "           ^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 815, in urlopen\n",
      "    return self.urlopen(\n",
      "           ^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 787, in urlopen\n",
      "    retries = retries.increment(\n",
      "              ^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\util\\retry.py\", line 592, in increment\n",
      "    raise MaxRetryError(_pool, url, error or ResponseError(cause))\n",
      "urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /search?q=Augsburg%2C+Bayern&format=json&limit=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000018FC1B814D0>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\adapters.py\", line 482, in _request\n",
      "    resp = self.session.get(url, timeout=timeout, headers=headers)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\sessions.py\", line 600, in get\n",
      "    return self.request(\"GET\", url, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\sessions.py\", line 587, in request\n",
      "    resp = self.send(prep, **send_kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\sessions.py\", line 701, in send\n",
      "    r = adapter.send(request, **kwargs)\n",
      "        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\adapters.py\", line 553, in send\n",
      "    raise ConnectTimeout(e, request=request)\n",
      "requests.exceptions.ConnectTimeout: HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /search?q=Augsburg%2C+Bayern&format=json&limit=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000018FC1B814D0>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\extra\\rate_limiter.py\", line 136, in _retries_gen\n",
      "    yield i  # Run the function.\n",
      "    ^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\extra\\rate_limiter.py\", line 274, in __call__\n",
      "    res = self.func(*args, **kwargs)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\geocoders\\nominatim.py\", line 297, in geocode\n",
      "    return self._call_geocoder(url, callback, timeout=timeout)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\geocoders\\base.py\", line 368, in _call_geocoder\n",
      "    result = self.adapter.get_json(url, timeout=timeout, headers=req_headers)\n",
      "             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\adapters.py\", line 472, in get_json\n",
      "    resp = self._request(url, timeout=timeout, headers=headers)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\adapters.py\", line 494, in _request\n",
      "    raise GeocoderUnavailable(message)\n",
      "geopy.exc.GeocoderUnavailable: HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /search?q=Augsburg%2C+Bayern&format=json&limit=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000018FC1B814D0>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "RateLimiter caught an error, retrying (0/2 tries). Called with (*('Frankfurt, Hessen',), **{}).\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connection.py\", line 174, in _new_conn\n",
      "    conn = connection.create_connection(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\util\\connection.py\", line 95, in create_connection\n",
      "    raise err\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\util\\connection.py\", line 85, in create_connection\n",
      "    sock.connect(sa)\n",
      "TimeoutError: timed out\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 703, in urlopen\n",
      "    httplib_response = self._make_request(\n",
      "                       ^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 386, in _make_request\n",
      "    self._validate_conn(conn)\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 1042, in _validate_conn\n",
      "    conn.connect()\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connection.py\", line 358, in connect\n",
      "    self.sock = conn = self._new_conn()\n",
      "                       ^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connection.py\", line 179, in _new_conn\n",
      "    raise ConnectTimeoutError(\n",
      "urllib3.exceptions.ConnectTimeoutError: (<urllib3.connection.HTTPSConnection object at 0x0000018F8A9F9BD0>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)')\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\adapters.py\", line 489, in send\n",
      "    resp = conn.urlopen(\n",
      "           ^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 815, in urlopen\n",
      "    return self.urlopen(\n",
      "           ^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 815, in urlopen\n",
      "    return self.urlopen(\n",
      "           ^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 787, in urlopen\n",
      "    retries = retries.increment(\n",
      "              ^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\util\\retry.py\", line 592, in increment\n",
      "    raise MaxRetryError(_pool, url, error or ResponseError(cause))\n",
      "urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /search?q=Frankfurt%2C+Hessen&format=json&limit=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000018F8A9F9BD0>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\adapters.py\", line 482, in _request\n",
      "    resp = self.session.get(url, timeout=timeout, headers=headers)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\sessions.py\", line 600, in get\n",
      "    return self.request(\"GET\", url, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\sessions.py\", line 587, in request\n",
      "    resp = self.send(prep, **send_kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\sessions.py\", line 701, in send\n",
      "    r = adapter.send(request, **kwargs)\n",
      "        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\adapters.py\", line 553, in send\n",
      "    raise ConnectTimeout(e, request=request)\n",
      "requests.exceptions.ConnectTimeout: HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /search?q=Frankfurt%2C+Hessen&format=json&limit=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000018F8A9F9BD0>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\extra\\rate_limiter.py\", line 136, in _retries_gen\n",
      "    yield i  # Run the function.\n",
      "    ^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\extra\\rate_limiter.py\", line 274, in __call__\n",
      "    res = self.func(*args, **kwargs)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\geocoders\\nominatim.py\", line 297, in geocode\n",
      "    return self._call_geocoder(url, callback, timeout=timeout)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\geocoders\\base.py\", line 368, in _call_geocoder\n",
      "    result = self.adapter.get_json(url, timeout=timeout, headers=req_headers)\n",
      "             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\adapters.py\", line 472, in get_json\n",
      "    resp = self._request(url, timeout=timeout, headers=headers)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\adapters.py\", line 494, in _request\n",
      "    raise GeocoderUnavailable(message)\n",
      "geopy.exc.GeocoderUnavailable: HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /search?q=Frankfurt%2C+Hessen&format=json&limit=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000018F8A9F9BD0>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "RateLimiter caught an error, retrying (1/2 tries). Called with (*('Frankfurt, Hessen',), **{}).\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connection.py\", line 174, in _new_conn\n",
      "    conn = connection.create_connection(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\util\\connection.py\", line 95, in create_connection\n",
      "    raise err\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\util\\connection.py\", line 85, in create_connection\n",
      "    sock.connect(sa)\n",
      "TimeoutError: timed out\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 703, in urlopen\n",
      "    httplib_response = self._make_request(\n",
      "                       ^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 386, in _make_request\n",
      "    self._validate_conn(conn)\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 1042, in _validate_conn\n",
      "    conn.connect()\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connection.py\", line 358, in connect\n",
      "    self.sock = conn = self._new_conn()\n",
      "                       ^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connection.py\", line 179, in _new_conn\n",
      "    raise ConnectTimeoutError(\n",
      "urllib3.exceptions.ConnectTimeoutError: (<urllib3.connection.HTTPSConnection object at 0x0000018F8A7EC390>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)')\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\adapters.py\", line 489, in send\n",
      "    resp = conn.urlopen(\n",
      "           ^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 815, in urlopen\n",
      "    return self.urlopen(\n",
      "           ^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 815, in urlopen\n",
      "    return self.urlopen(\n",
      "           ^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 787, in urlopen\n",
      "    retries = retries.increment(\n",
      "              ^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\util\\retry.py\", line 592, in increment\n",
      "    raise MaxRetryError(_pool, url, error or ResponseError(cause))\n",
      "urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /search?q=Frankfurt%2C+Hessen&format=json&limit=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000018F8A7EC390>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\adapters.py\", line 482, in _request\n",
      "    resp = self.session.get(url, timeout=timeout, headers=headers)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\sessions.py\", line 600, in get\n",
      "    return self.request(\"GET\", url, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\sessions.py\", line 587, in request\n",
      "    resp = self.send(prep, **send_kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\sessions.py\", line 701, in send\n",
      "    r = adapter.send(request, **kwargs)\n",
      "        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\adapters.py\", line 553, in send\n",
      "    raise ConnectTimeout(e, request=request)\n",
      "requests.exceptions.ConnectTimeout: HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /search?q=Frankfurt%2C+Hessen&format=json&limit=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000018F8A7EC390>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\extra\\rate_limiter.py\", line 136, in _retries_gen\n",
      "    yield i  # Run the function.\n",
      "    ^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\extra\\rate_limiter.py\", line 274, in __call__\n",
      "    res = self.func(*args, **kwargs)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\geocoders\\nominatim.py\", line 297, in geocode\n",
      "    return self._call_geocoder(url, callback, timeout=timeout)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\geocoders\\base.py\", line 368, in _call_geocoder\n",
      "    result = self.adapter.get_json(url, timeout=timeout, headers=req_headers)\n",
      "             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\adapters.py\", line 472, in get_json\n",
      "    resp = self._request(url, timeout=timeout, headers=headers)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\algro\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\geopy\\adapters.py\", line 494, in _request\n",
      "    raise GeocoderUnavailable(message)\n",
      "geopy.exc.GeocoderUnavailable: HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /search?q=Frankfurt%2C+Hessen&format=json&limit=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000018F8A7EC390>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n"
     ]
    }
   ],
   "source": [
    "coord_dict = {}\n",
    "\n",
    "# Initialize geolocator\n",
    "geolocator = Nominatim(user_agent=\"geoapiExercises\")\n",
    "\n",
    "# Use rate limiter to not hit service limits\n",
    "geocode = RateLimiter(geolocator.geocode, min_delay_seconds=1)\n",
    "\n",
    "# Function to get coordinates\n",
    "def fetch_coordinates(place):\n",
    "    try:\n",
    "        location = geocode(place)\n",
    "        return pd.Series([location.latitude, location.longitude])\n",
    "    except AttributeError:\n",
    "        return pd.Series([None, None])\n",
    "\n",
    "for place in places:\n",
    "    location = fetch_coordinates(place)\n",
    "    coord_dict[place] = [location.iloc[0], location.iloc[1]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'Erfurt, Thüringen': [50.9777974, 11.0287364], 'Rudolstadt, Thüringen': [50.7206063, 11.3401985], 'Augsburg, Bayern': [48.3690341, 10.8979522], 'Frankfurt, Hessen': [50.1106444, 8.6820917], 'Würzburg, Bayern': [49.7933723, 9.9309779], 'Bayreuth, Bayern': [49.9427202, 11.5763079], 'Berlin, Preussen': [52.517253, 13.3927431], 'Breslau, Schlesien': [51.1263106, 16.97819633051261], 'Leipzig, Sachsen': [51.3406321, 12.3747329], 'Bremen': [53.0758196, 8.8071646], 'Mainz': [50.0012314, 8.2762513], 'Stuttgart, Württemberg': [48.7820489, 9.2687566], 'Hattingen, Nordrhein-Westfalen': [51.4007175, 7.1862486], 'Berlin': [52.5170365, 13.3888599], 'Tübingen, Württemberg': [48.5236164, 9.0535531], 'Wien, Österreich': [48.2083537, 16.3725042]}\n"
     ]
    }
   ],
   "source": [
    "print(coord_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\algro\\AppData\\Local\\Temp\\ipykernel_25312\\1757184783.py:9: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_new[['latitude', 'longitude']] = df_new['Place_corrected'].apply(get_coordinates).apply(pd.Series)\n",
      "C:\\Users\\algro\\AppData\\Local\\Temp\\ipykernel_25312\\1757184783.py:9: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_new[['latitude', 'longitude']] = df_new['Place_corrected'].apply(get_coordinates).apply(pd.Series)\n"
     ]
    }
   ],
   "source": [
    "# Add Latitude and Longitude to DataFrame \n",
    "def get_coordinates(place):\n",
    "    if place in coord_dict:\n",
    "        return pd.Series(coord_dict[place])\n",
    "    else:\n",
    "        return pd.Series([None, None])\n",
    "\n",
    "df_new[['latitude', 'longitude']] = df_new['Place_corrected'].apply(get_coordinates).apply(pd.Series)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Date</th>\n",
       "      <th>Place</th>\n",
       "      <th>Content</th>\n",
       "      <th>Words</th>\n",
       "      <th>Year</th>\n",
       "      <th>Day_month</th>\n",
       "      <th>Day</th>\n",
       "      <th>Month</th>\n",
       "      <th>Place_corrected</th>\n",
       "      <th>latitude</th>\n",
       "      <th>longitude</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>khz01001</td>\n",
       "      <td>1744-01-10</td>\n",
       "      <td>Erfurt, Thüringen</td>\n",
       "      <td>Der allerneuesten \\n\\nEuropäischen \\n\\nWelt- u...</td>\n",
       "      <td>[(Clevischen), (Graf, von), (Holland), (hertzl...</td>\n",
       "      <td>1744</td>\n",
       "      <td>10-01</td>\n",
       "      <td>10</td>\n",
       "      <td>01</td>\n",
       "      <td>Erfurt, Thüringen</td>\n",
       "      <td>50.977797</td>\n",
       "      <td>11.028736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>khz01002</td>\n",
       "      <td>1744-01-13</td>\n",
       "      <td>Erfurt, Thüringen</td>\n",
       "      <td>Der allerneuesten   Europäischen   Welt- und S...</td>\n",
       "      <td>[(Fürst, von, Lichtenstein), (Ubergangs), (Eng...</td>\n",
       "      <td>1744</td>\n",
       "      <td>13-01</td>\n",
       "      <td>13</td>\n",
       "      <td>01</td>\n",
       "      <td>Erfurt, Thüringen</td>\n",
       "      <td>50.977797</td>\n",
       "      <td>11.028736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>khz01003</td>\n",
       "      <td>1744-01-17</td>\n",
       "      <td>Erfurt, Thüringen</td>\n",
       "      <td>Der allerneuesten   Europäischen   Welt- und S...</td>\n",
       "      <td>[(Amts,  ), (Holland), (Jtalien), (August, von...</td>\n",
       "      <td>1744</td>\n",
       "      <td>17-01</td>\n",
       "      <td>17</td>\n",
       "      <td>01</td>\n",
       "      <td>Erfurt, Thüringen</td>\n",
       "      <td>50.977797</td>\n",
       "      <td>11.028736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>khz01004</td>\n",
       "      <td>1744-02-07</td>\n",
       "      <td>Erfurt, Thüringen</td>\n",
       "      <td>Der allerneuesten   Europäischen   Welt- und S...</td>\n",
       "      <td>[(Po), (Jhren, Sold), (Chur), (Pescaro), (D), ...</td>\n",
       "      <td>1744</td>\n",
       "      <td>07-02</td>\n",
       "      <td>07</td>\n",
       "      <td>02</td>\n",
       "      <td>Erfurt, Thüringen</td>\n",
       "      <td>50.977797</td>\n",
       "      <td>11.028736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>khz01005</td>\n",
       "      <td>1744-02-21</td>\n",
       "      <td>Erfurt, Thüringen</td>\n",
       "      <td>Der allerneuesten   Europäischen   Welt-und St...</td>\n",
       "      <td>[(Frantzösischen, Escadern), (Canonicum), (hof...</td>\n",
       "      <td>1744</td>\n",
       "      <td>21-02</td>\n",
       "      <td>21</td>\n",
       "      <td>02</td>\n",
       "      <td>Erfurt, Thüringen</td>\n",
       "      <td>50.977797</td>\n",
       "      <td>11.028736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>647</th>\n",
       "      <td>khz21009</td>\n",
       "      <td>1850-10-12</td>\n",
       "      <td>Wien, Österreich</td>\n",
       "      <td>Wiener    Zeitung. \\n \\n   N ro   244.   Samst...</td>\n",
       "      <td>[(Hoheiten), (viribus), (Joseph,  , Schütz), (...</td>\n",
       "      <td>1850</td>\n",
       "      <td>12-10</td>\n",
       "      <td>12</td>\n",
       "      <td>10</td>\n",
       "      <td>Wien, Österreich</td>\n",
       "      <td>48.208354</td>\n",
       "      <td>16.372504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>648</th>\n",
       "      <td>khz21010</td>\n",
       "      <td>1850-10-17</td>\n",
       "      <td>Wien, Österreich</td>\n",
       "      <td>Wiener    Zeitung. \\n \\n   N ro   248.   Donne...</td>\n",
       "      <td>[(Besitzanfällen), (Brixen), (Anschlusse), (Ba...</td>\n",
       "      <td>1850</td>\n",
       "      <td>17-10</td>\n",
       "      <td>17</td>\n",
       "      <td>10</td>\n",
       "      <td>Wien, Österreich</td>\n",
       "      <td>48.208354</td>\n",
       "      <td>16.372504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>649</th>\n",
       "      <td>khz21011</td>\n",
       "      <td>1850-10-25</td>\n",
       "      <td>Wien, Österreich</td>\n",
       "      <td>Wiener    Zeitung. \\n \\n   N ro   255.   Freit...</td>\n",
       "      <td>[(Belgier), (Deutschen, Bundes), (Herzogthums,...</td>\n",
       "      <td>1850</td>\n",
       "      <td>25-10</td>\n",
       "      <td>25</td>\n",
       "      <td>10</td>\n",
       "      <td>Wien, Österreich</td>\n",
       "      <td>48.208354</td>\n",
       "      <td>16.372504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>650</th>\n",
       "      <td>khz21012</td>\n",
       "      <td>1850-10-29</td>\n",
       "      <td>Wien, Österreich</td>\n",
       "      <td>Wiener    Zeitung. \\n \\n   N ro   258   Dinsta...</td>\n",
       "      <td>[(Hrn., J.,  , Prettner), (FlügelAdjutanten,  ...</td>\n",
       "      <td>1850</td>\n",
       "      <td>29-10</td>\n",
       "      <td>29</td>\n",
       "      <td>10</td>\n",
       "      <td>Wien, Österreich</td>\n",
       "      <td>48.208354</td>\n",
       "      <td>16.372504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>651</th>\n",
       "      <td>khz21013</td>\n",
       "      <td>1850-10-31</td>\n",
       "      <td>Wien, Österreich</td>\n",
       "      <td>Wiener    Zeitung. \\n \\n   N ro   260   Donner...</td>\n",
       "      <td>[(England), (Dänemark), (Himalaya), (Matzleins...</td>\n",
       "      <td>1850</td>\n",
       "      <td>31-10</td>\n",
       "      <td>31</td>\n",
       "      <td>10</td>\n",
       "      <td>Wien, Österreich</td>\n",
       "      <td>48.208354</td>\n",
       "      <td>16.372504</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>596 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           ID       Date              Place  \\\n",
       "0    khz01001 1744-01-10  Erfurt, Thüringen   \n",
       "1    khz01002 1744-01-13  Erfurt, Thüringen   \n",
       "2    khz01003 1744-01-17  Erfurt, Thüringen   \n",
       "3    khz01004 1744-02-07  Erfurt, Thüringen   \n",
       "4    khz01005 1744-02-21  Erfurt, Thüringen   \n",
       "..        ...        ...                ...   \n",
       "647  khz21009 1850-10-12   Wien, Österreich   \n",
       "648  khz21010 1850-10-17   Wien, Österreich   \n",
       "649  khz21011 1850-10-25   Wien, Österreich   \n",
       "650  khz21012 1850-10-29   Wien, Österreich   \n",
       "651  khz21013 1850-10-31   Wien, Österreich   \n",
       "\n",
       "                                               Content  \\\n",
       "0    Der allerneuesten \\n\\nEuropäischen \\n\\nWelt- u...   \n",
       "1    Der allerneuesten   Europäischen   Welt- und S...   \n",
       "2    Der allerneuesten   Europäischen   Welt- und S...   \n",
       "3    Der allerneuesten   Europäischen   Welt- und S...   \n",
       "4    Der allerneuesten   Europäischen   Welt-und St...   \n",
       "..                                                 ...   \n",
       "647  Wiener    Zeitung. \\n \\n   N ro   244.   Samst...   \n",
       "648  Wiener    Zeitung. \\n \\n   N ro   248.   Donne...   \n",
       "649  Wiener    Zeitung. \\n \\n   N ro   255.   Freit...   \n",
       "650  Wiener    Zeitung. \\n \\n   N ro   258   Dinsta...   \n",
       "651  Wiener    Zeitung. \\n \\n   N ro   260   Donner...   \n",
       "\n",
       "                                                 Words  Year Day_month Day  \\\n",
       "0    [(Clevischen), (Graf, von), (Holland), (hertzl...  1744     10-01  10   \n",
       "1    [(Fürst, von, Lichtenstein), (Ubergangs), (Eng...  1744     13-01  13   \n",
       "2    [(Amts,  ), (Holland), (Jtalien), (August, von...  1744     17-01  17   \n",
       "3    [(Po), (Jhren, Sold), (Chur), (Pescaro), (D), ...  1744     07-02  07   \n",
       "4    [(Frantzösischen, Escadern), (Canonicum), (hof...  1744     21-02  21   \n",
       "..                                                 ...   ...       ...  ..   \n",
       "647  [(Hoheiten), (viribus), (Joseph,  , Schütz), (...  1850     12-10  12   \n",
       "648  [(Besitzanfällen), (Brixen), (Anschlusse), (Ba...  1850     17-10  17   \n",
       "649  [(Belgier), (Deutschen, Bundes), (Herzogthums,...  1850     25-10  25   \n",
       "650  [(Hrn., J.,  , Prettner), (FlügelAdjutanten,  ...  1850     29-10  29   \n",
       "651  [(England), (Dänemark), (Himalaya), (Matzleins...  1850     31-10  31   \n",
       "\n",
       "    Month    Place_corrected   latitude  longitude  \n",
       "0      01  Erfurt, Thüringen  50.977797  11.028736  \n",
       "1      01  Erfurt, Thüringen  50.977797  11.028736  \n",
       "2      01  Erfurt, Thüringen  50.977797  11.028736  \n",
       "3      02  Erfurt, Thüringen  50.977797  11.028736  \n",
       "4      02  Erfurt, Thüringen  50.977797  11.028736  \n",
       "..    ...                ...        ...        ...  \n",
       "647    10   Wien, Österreich  48.208354  16.372504  \n",
       "648    10   Wien, Österreich  48.208354  16.372504  \n",
       "649    10   Wien, Österreich  48.208354  16.372504  \n",
       "650    10   Wien, Österreich  48.208354  16.372504  \n",
       "651    10   Wien, Österreich  48.208354  16.372504  \n",
       "\n",
       "[596 rows x 12 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_new"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Exporting the Final DataFrame\n",
    "df_new.to_csv('Tei_dataframe without MISC.csv', index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
